{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "53ab4db9-a426-4a1e-b229-57dde964d9eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "bp = \"PRJNA665227\"\n",
    "fastq_folder = f\"/home/laura/projects/virus-watch-data/{bp}/bwa/raw\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "24129e9c-cd0c-47db-af27-489453f6dedb",
   "metadata": {},
   "outputs": [],
   "source": [
    "virus_fasta = \"/home/laura/projects/virus-watch-data/virus_ref/uniques_noduplicates.fa\"\n",
    "virus_index = \"/home/laura/projects/virus-watch-data/virus_ref/kallisto_aa/noduplicates.idx\"\n",
    "virus_t2g = \"/home/laura/projects/virus-watch-data/virus_ref/nodup_clu_t2g.txt\"\n",
    "\n",
    "out_folder = f\"/home/laura/projects/virus-watch-data/{bp}/full_index_bwa\"\n",
    "host_out_folder = f\"/home/laura/projects/virus-watch-data/{bp}/full_index_stringent_dlist_flank1/host_canine\"\n",
    "\n",
    "kallisto = \"/home/laura/projects/kallisto/build/src/kallisto\"\n",
    "bustools = \"/home/laura/projects/bustools/build/src/bustools\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb590f46-fb15-4289-8dc1-0fc0dd658958",
   "metadata": {},
   "source": [
    "Create new batch file with paths to files where host reads were removed based on bwa alignment:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "98065f87-cd35-4275-8bec-eba9728f5e14",
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_batch_file = f\"{fastq_folder}/batch.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "52cf5c77-70bb-4f33-b5ff-4f167c11c0bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "1b88ef1c-e15e-47b6-81a9-0225cadc9824",
   "metadata": {},
   "outputs": [],
   "source": [
    "fastqs = []\n",
    "for filename in glob.glob(f\"{fastq_folder}/*.fastq\"):\n",
    "    fastqs.append(filename.split(\"/\")[-1])\n",
    "    \n",
    "fastqs.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "1db8bbb5-43cd-43f2-9e0f-2f846ef0661c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "212"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fastqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "984b4db6-3251-4511-9936-58ab0c880c99",
   "metadata": {},
   "outputs": [],
   "source": [
    "samples = []\n",
    "for fastq in fastqs:\n",
    "    samples.append(fastq.split(\"_\")[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "6c97e6b6-6557-4964-9b19-eaf9d20acd46",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "106"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samples = list(set(samples))\n",
    "len(samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6e2836ff-7126-4290-8259-08806b4796b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(sample_batch_file, \"w\") as batch_file:\n",
    "    for sample in samples:\n",
    "        fastq1 = sample + \"_1.fastq\"\n",
    "        fastq2 = sample + \"_2.fastq\"\n",
    "        batch_file.write(sample + \"\\t\" + fastq_folder + \"/\" + fastq1 + \"\\t\" + fastq_folder + \"/\" + fastq2 + \"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f0353728-0f36-45f9-8f68-830c086de61d",
   "metadata": {},
   "source": [
    "Generate virus reference index without dlist:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "808c42e0-67b6-4a4c-a4f5-e7da96c06957",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[index] WARNING --d-list-overhang set to < 3; should be >= 3 with --aa\n",
      "[build] loading fasta file /home/laura/projects/virus-watch-data/virus_ref/uniques_noduplicates.fa\n",
      "[build] k-mer length: 31\n",
      "KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
      "KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
      "KmerStream::KmerStream(): Finished\n",
      "CompactedDBG::build(): Estimated number of k-mers occurring at least once: 37641510\n",
      "CompactedDBG::build(): Estimated number of minimizer occurring at least once: 7877811\n",
      "CompactedDBG::filter(): Processed 87630084 k-mers in 296561 reads\n",
      "CompactedDBG::filter(): Found 37508917 unique k-mers\n",
      "CompactedDBG::filter(): Number of blocks in Bloom filter is 257317\n",
      "CompactedDBG::construct(): Extract approximate unitigs (1/2)\n",
      "CompactedDBG::construct(): Extract approximate unitigs (2/2)\n",
      "CompactedDBG::construct(): Closed all input files\n",
      "\n",
      "CompactedDBG::construct(): Splitting unitigs (1/2)\n",
      "\n",
      "CompactedDBG::construct(): Splitting unitigs (2/2)\n",
      "CompactedDBG::construct(): Before split: 2040432 unitigs\n",
      "CompactedDBG::construct(): After split (1/1): 2040432 unitigs\n",
      "CompactedDBG::construct(): Unitigs split: 1631\n",
      "CompactedDBG::construct(): Unitigs deleted: 0\n",
      "\n",
      "CompactedDBG::construct(): Joining unitigs\n",
      "CompactedDBG::construct(): After join: 2021701 unitigs\n",
      "CompactedDBG::construct(): Joined 19096 unitigs\n",
      "[build] building MPHF\n",
      "[build] creating equivalence classes ... \n",
      "[build] target de Bruijn graph has k-mer length 31 and minimizer length 23\n",
      "[build] target de Bruijn graph has 2021701 contigs and contains 37541756 k-mers \n",
      "\n",
      "\tCommand being timed: \"/home/laura/projects/kallisto/build/src/kallisto index -t 30 --aa -i /home/laura/projects/virus-watch-data/virus_ref/kallisto_aa/noduplicates.idx /home/laura/projects/virus-watch-data/virus_ref/uniques_noduplicates.fa\"\n",
      "\tUser time (seconds): 877.36\n",
      "\tSystem time (seconds): 9.82\n",
      "\tPercent of CPU this job got: 905%\n",
      "\tElapsed (wall clock) time (h:mm:ss or m:ss): 1:38.02\n",
      "\tAverage shared text size (kbytes): 0\n",
      "\tAverage unshared data size (kbytes): 0\n",
      "\tAverage stack size (kbytes): 0\n",
      "\tAverage total size (kbytes): 0\n",
      "\tMaximum resident set size (kbytes): 4393832\n",
      "\tAverage resident set size (kbytes): 0\n",
      "\tMajor (requiring I/O) page faults: 1\n",
      "\tMinor (reclaiming a frame) page faults: 2775483\n",
      "\tVoluntary context switches: 102641\n",
      "\tInvoluntary context switches: 1545\n",
      "\tSwaps: 0\n",
      "\tFile system inputs: 72816\n",
      "\tFile system outputs: 1407400\n",
      "\tSocket messages sent: 0\n",
      "\tSocket messages received: 0\n",
      "\tSignals delivered: 0\n",
      "\tPage size (bytes): 4096\n",
      "\tExit status: 0\n"
     ]
    }
   ],
   "source": [
    "!/usr/bin/time -v $kallisto index \\\n",
    "    -t 30 \\\n",
    "    --aa \\\n",
    "    -i $virus_index \\\n",
    "    $virus_fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3dfe6aa8-11ad-4391-bc96-cb0bed9c92e7",
   "metadata": {},
   "source": [
    "Align files and correct barcodes based on host whitelist:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "21b42c02-8149-4ec4-91e3-0d92a4ece7a1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[bus] will try running read files supplied in batch file\n",
      "[bus] Note: Strand option was not specified; setting it to --unstranded for specified technology\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,756\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698573_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698573_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698501_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698501_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698523_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698523_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698521_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698521_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698566_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698566_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698519_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698519_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698529_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698529_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698533_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698533_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698568_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698568_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698539_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698539_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698574_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698574_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698506_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698506_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698514_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698514_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698512_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698512_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698510_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698510_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698538_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698538_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698579_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698579_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698545_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698545_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698554_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698554_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698580_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698580_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698560_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698560_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698571_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698571_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698601_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698601_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698582_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698582_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698603_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698603_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698518_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698518_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698511_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698511_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698504_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698504_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698598_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698598_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698507_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698507_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698594_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698594_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698585_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698585_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698555_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698555_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698508_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698508_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698587_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698587_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698591_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698591_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698581_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698581_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698515_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698515_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698537_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698537_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698548_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698548_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698590_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698590_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698589_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698589_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698569_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698569_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698572_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698572_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698543_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698543_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698570_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698570_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698531_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698531_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698576_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698576_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698525_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698525_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698532_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698532_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698578_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698578_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698567_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698567_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698522_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698522_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698558_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698558_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698542_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698542_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698528_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698528_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698593_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698593_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698584_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698584_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698549_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698549_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698505_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698505_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698561_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698561_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698600_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698600_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698583_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698583_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698550_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698550_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698524_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698524_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698499_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698499_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698556_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698556_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698553_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698553_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698527_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698527_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698541_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698541_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698544_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698544_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698604_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698604_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698546_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698546_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698602_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698602_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698595_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698595_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698563_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698563_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698551_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698551_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698586_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698586_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698562_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698562_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698502_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698502_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698535_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698535_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698599_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698599_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698500_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698500_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698559_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698559_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698564_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698564_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698565_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698565_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698536_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698536_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698577_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698577_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698517_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698517_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698592_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698592_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698596_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698596_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698547_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698547_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698552_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698552_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698526_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698526_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698503_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698503_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698575_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698575_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698513_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698513_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698530_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698530_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698588_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698588_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698534_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698534_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698509_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698509_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698557_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698557_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698597_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698597_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698516_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698516_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698540_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698540_2.fastq\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698520_1.fastq\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/SRR12698520_2.fastq\n",
      "[progress] 2339M reads processed (10.0% mapped)             .4% mapped)             11.6% mapped)             11.0% mapped)             10.7% mapped)             10.5% mapped)             10.5% mapped)             11.0% mapped)              done\n",
      "[quant] processed 2,340,332,547 reads, 234,424,610 reads pseudoaligned\n",
      "\n",
      "\tCommand being timed: \"/home/laura/projects/kallisto/build/src/kallisto bus -i /home/laura/projects/virus-watch-data/virus_ref/kallisto_aa/noduplicates.idx -o /home/laura/projects/virus-watch-data/PRJNA665227/full_index_bwa --aa -t 30 -B /home/laura/projects/virus-watch-data/PRJNA665227/bwa/raw/batch.txt --batch-barcodes -x 0,0,12:0,12,20:1,0,0\"\n",
      "\tUser time (seconds): 269373.52\n",
      "\tSystem time (seconds): 730.34\n",
      "\tPercent of CPU this job got: 2985%\n",
      "\tElapsed (wall clock) time (h:mm:ss or m:ss): 2:30:46\n",
      "\tAverage shared text size (kbytes): 0\n",
      "\tAverage unshared data size (kbytes): 0\n",
      "\tAverage stack size (kbytes): 0\n",
      "\tAverage total size (kbytes): 0\n",
      "\tMaximum resident set size (kbytes): 1553440\n",
      "\tAverage resident set size (kbytes): 0\n",
      "\tMajor (requiring I/O) page faults: 3\n",
      "\tMinor (reclaiming a frame) page faults: 69556021\n",
      "\tVoluntary context switches: 458966\n",
      "\tInvoluntary context switches: 823074\n",
      "\tSwaps: 0\n",
      "\tFile system inputs: 1308775944\n",
      "\tFile system outputs: 14873720\n",
      "\tSocket messages sent: 0\n",
      "\tSocket messages received: 0\n",
      "\tSignals delivered: 0\n",
      "\tPage size (bytes): 4096\n",
      "\tExit status: 0\n"
     ]
    }
   ],
   "source": [
    "!/usr/bin/time -v $kallisto bus \\\n",
    "        -i $virus_index \\\n",
    "        -o $out_folder \\\n",
    "        --aa \\\n",
    "        -t 30 \\\n",
    "        -B $sample_batch_file \\\n",
    "        --batch-barcodes \\\n",
    "        -x 0,0,12:0,12,20:1,0,0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "0fb6a377-5005-499c-8292-71069d14f1b2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "partition time: 3.74s\n",
      "partition time: 2.21s\n",
      "Read in 234424610 BUS records\n",
      "reading time 4.44s\n",
      "sorting time 49.3s\n",
      "writing time 0s\n",
      "Found 775516 barcodes in the on-list\n",
      "Processed 65542131 BUS records\n",
      "In on-list = 13464785\n",
      "Corrected    = 15472040\n",
      "Uncorrected  = 36605306\n",
      "partition time: 0.48s\n",
      " all fits in buffer\n",
      "Read in 28936825 BUS records\n",
      "reading time 0.41s\n",
      "sorting time 5.57s\n",
      "writing time 0.85s\n",
      "CPU times: user 1.85 s, sys: 378 ms, total: 2.23 s\n",
      "Wall time: 1min 53s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!$bustools sort \\\n",
    "    -m 4G \\\n",
    "    -t 20 \\\n",
    "    -o $out_folder/output_sorted.bus \\\n",
    "    $out_folder/output.bus\n",
    "\n",
    "!$bustools correct \\\n",
    "    -w $host_out_folder/bustools_whitelist.txt \\\n",
    "    -o $out_folder/output_sorted_corrected.bus \\\n",
    "    $out_folder/output_sorted.bus\n",
    "\n",
    "!$bustools sort \\\n",
    "    -m 4G \\\n",
    "    -t 30 \\\n",
    "    -o $out_folder/output_sorted_corrected_sorted.bus \\\n",
    "    $out_folder/output_sorted_corrected.bus\n",
    "\n",
    "!$bustools count \\\n",
    "    --genecounts \\\n",
    "    -o $out_folder/bustools_count/ \\\n",
    "    -g $virus_t2g \\\n",
    "    -e $out_folder/matrix.ec \\\n",
    "    -t $out_folder/transcripts.txt \\\n",
    "    $out_folder/output_sorted_corrected_sorted.bus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f92ad4b4-171b-4cc1-82b8-2238521e2167",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
