{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e8ef6419-dd14-4729-9ea2-cb9fa4796794",
   "metadata": {},
   "source": [
    "# Align to combined macaque and canine reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a4824ed9-3ed9-4bea-9142-0325580190be",
   "metadata": {},
   "outputs": [],
   "source": [
    "bp = \"PRJNA665227\"\n",
    "fastq_folder = f\"/home/laura/projects/virus-watch-data/{bp}/raw\"\n",
    "sample_batch_file = f\"{fastq_folder}/batch.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01c3dfa9-d995-4512-b71d-db4351e28053",
   "metadata": {},
   "source": [
    "Generate batch file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "560c5f4b-871c-420f-ae05-846ac2048104",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c7fa30fa-8b32-4ffc-a772-f2b7c3410921",
   "metadata": {},
   "outputs": [],
   "source": [
    "fastqs = []\n",
    "for filename in glob.glob(f\"{fastq_folder}/*.fastq.gz\"):\n",
    "    fastqs.append(filename.split(\"/\")[-1])\n",
    "    \n",
    "fastqs.sort()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "baa65c7c-248e-4aae-89a9-90cf2c6df1d2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "212"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(fastqs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "cb0c33d4-59fc-4c4b-8402-40596e11f0b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# samples = []\n",
    "# for fastq in fastqs:\n",
    "#     samples.append(fastq.split(\"_\")[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "79a9a16f-ca9e-463e-9821-4e0a196809ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "# samples = list(set(samples))\n",
    "# len(samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "4718189b-ee28-4daa-92ab-60f46a6e01cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# with open(sample_batch_file, \"w\") as batch_file:\n",
    "#     for sample in samples:\n",
    "#         fastq1 = sample + \"_1.fastq.gz\"\n",
    "#         fastq2 = sample + \"_2.fastq.gz\"\n",
    "#         batch_file.write(sample + \"\\t\" + fastq_folder + \"/\" + fastq1 + \"\\t\" + fastq_folder + \"/\" + fastq2 + \"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8af1e70-b630-4b61-b967-4257d85a28da",
   "metadata": {},
   "source": [
    "### Generate combined macaque / canine reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "09c83abf-5137-443e-b9df-1f7066110955",
   "metadata": {},
   "outputs": [],
   "source": [
    "host_fasta = \"/home/laura/projects/virus-watch-data/rhesus_ref/Macaca_mulatta.Mmul_10.dna.toplevel.fa.gz\"\n",
    "host_gtf = \"/home/laura/projects/virus-watch-data/rhesus_ref/Macaca_mulatta.Mmul_10.109.gtf.gz\"\n",
    "\n",
    "canine_fasta = \"/home/laura/projects/virus-watch-data/canine_ref/Canis_lupus_familiaris.ROS_Cfam_1.0.dna.toplevel.fa.gz\"\n",
    "canine_gtf = \"/home/laura/projects/virus-watch-data/canine_ref/Canis_lupus_familiaris.ROS_Cfam_1.0.109.gtf.gz\"\n",
    "\n",
    "canine_macaque_fasta = \"/home/laura/projects/virus-watch-data/canine_macaque_ref/combined_genomes.dna.toplevel.fa.gz\"\n",
    "\n",
    "index = \"/home/laura/projects/virus-watch-data/canine_macaque_ref/index.idx\"\n",
    "t2g = \"/home/laura/projects/virus-watch-data/canine_macaque_ref/t2g.txt\"\n",
    "fa = \"/home/laura/projects/virus-watch-data/canine_macaque_ref/transcriptome.fa\"\n",
    "\n",
    "kallisto = \"/home/laura/projects/kallisto/build/src/kallisto\"\n",
    "bustools = \"/home/laura/projects/bustools/build/src/bustools\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9dc5959a-5b9f-4c0e-a342-4b02533b2587",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2023-07-05 13:06:12,688]   DEBUG [main] Printing verbose output\n",
      "[2023-07-05 13:06:14,920]   DEBUG [main] kallisto binary located at /home/laura/projects/kallisto/build/src/kallisto\n",
      "[2023-07-05 13:06:14,920]   DEBUG [main] bustools binary located at /home/laura/projects/bustools/build/src/bustools\n",
      "[2023-07-05 13:06:14,920]   DEBUG [main] Creating `tmp` directory\n",
      "[2023-07-05 13:06:14,931]   DEBUG [main] Namespace(list=False, command='ref', tmp=None, keep_tmp=False, verbose=True, i='/home/laura/projects/virus-watch-data/canine_macaque_ref/index.idx', g='/home/laura/projects/virus-watch-data/canine_macaque_ref/t2g.txt', f1='/home/laura/projects/virus-watch-data/canine_macaque_ref/transcriptome.fa', include_attribute=None, exclude_attribute=None, f2=None, c1=None, c2=None, d=None, k=None, t=30, d_list=None, aa=False, workflow='standard', distinguish=False, make_unique=False, overwrite=True, kallisto='/home/laura/projects/kallisto/build/src/kallisto', bustools='/home/laura/projects/bustools/build/src/bustools', fasta='/home/laura/projects/virus-watch-data/canine_ref/Canis_lupus_familiaris.ROS_Cfam_1.0.dna.toplevel.fa.gz,/home/laura/projects/virus-watch-data/rhesus_ref/Macaca_mulatta.Mmul_10.dna.toplevel.fa.gz', gtf='/home/laura/projects/virus-watch-data/canine_ref/Canis_lupus_familiaris.ROS_Cfam_1.0.109.gtf.gz,/home/laura/projects/virus-watch-data/rhesus_ref/Macaca_mulatta.Mmul_10.109.gtf.gz', feature=None, no_mismatches=False, flank=None)\n",
      "[2023-07-05 13:06:14,931]    INFO [ref] Preparing /home/laura/projects/virus-watch-data/canine_ref/Canis_lupus_familiaris.ROS_Cfam_1.0.dna.toplevel.fa.gz, /home/laura/projects/virus-watch-data/canine_ref/Canis_lupus_familiaris.ROS_Cfam_1.0.109.gtf.gz\n",
      "[2023-07-05 13:06:43,394]    INFO [ref] Splitting genome /home/laura/projects/virus-watch-data/canine_ref/Canis_lupus_familiaris.ROS_Cfam_1.0.dna.toplevel.fa.gz into cDNA at /home/laura/projects/virus-watch/seqwell/ebola_PRJNA665227/tmp/tmpltq_9vak\n",
      "[2023-07-05 13:08:06,501]    INFO [ref] Preparing /home/laura/projects/virus-watch-data/rhesus_ref/Macaca_mulatta.Mmul_10.dna.toplevel.fa.gz, /home/laura/projects/virus-watch-data/rhesus_ref/Macaca_mulatta.Mmul_10.109.gtf.gz\n",
      "[2023-07-05 13:08:36,255]    INFO [ref] Splitting genome /home/laura/projects/virus-watch-data/rhesus_ref/Macaca_mulatta.Mmul_10.dna.toplevel.fa.gz into cDNA at /home/laura/projects/virus-watch/seqwell/ebola_PRJNA665227/tmp/tmp0wybh00r\n",
      "[2023-07-05 13:10:48,610]    INFO [ref] Concatenating 2 cDNAs to /home/laura/projects/virus-watch-data/canine_macaque_ref/transcriptome.fa\n",
      "[2023-07-05 13:10:49,279]    INFO [ref] Creating transcript-to-gene mapping at /home/laura/projects/virus-watch-data/canine_macaque_ref/t2g.txt\n",
      "[2023-07-05 13:10:50,517]    INFO [ref] Indexing /home/laura/projects/virus-watch-data/canine_macaque_ref/transcriptome.fa to /home/laura/projects/virus-watch-data/canine_macaque_ref/index.idx\n",
      "[2023-07-05 13:10:50,517]   DEBUG [ref] kallisto index -i /home/laura/projects/virus-watch-data/canine_macaque_ref/index.idx -k 31 -t 30 -d /home/laura/projects/virus-watch-data/canine_ref/Canis_lupus_familiaris.ROS_Cfam_1.0.dna.toplevel.fa.gz,/home/laura/projects/virus-watch-data/rhesus_ref/Macaca_mulatta.Mmul_10.dna.toplevel.fa.gz /home/laura/projects/virus-watch-data/canine_macaque_ref/transcriptome.fa\n",
      "[2023-07-05 13:10:50,659]   DEBUG [ref] \n",
      "[2023-07-05 13:10:50,659]   DEBUG [ref] [build] loading fasta file /home/laura/projects/virus-watch-data/canine_macaque_ref/transcriptome.fa\n",
      "[2023-07-05 13:10:50,659]   DEBUG [ref] [build] k-mer length: 31\n",
      "[2023-07-05 13:10:55,767]   DEBUG [ref] [build] warning: clipped off poly-A tail (longer than 10)\n",
      "[2023-07-05 13:10:55,767]   DEBUG [ref] from 5179 target sequences\n",
      "[2023-07-05 13:10:55,768]   DEBUG [ref] [build] warning: replaced 1339 non-ACGUT characters in the input sequence\n",
      "[2023-07-05 13:10:55,768]   DEBUG [ref] with pseudorandom nucleotides\n",
      "[2023-07-05 13:10:55,768]   DEBUG [ref] KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
      "[2023-07-05 13:10:55,768]   DEBUG [ref] KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
      "[2023-07-05 13:10:58,372]   DEBUG [ref] KmerStream::KmerStream(): Finished\n",
      "[2023-07-05 13:10:58,673]   DEBUG [ref] CompactedDBG::build(): Estimated number of k-mers occurring at least once: 146320997\n",
      "[2023-07-05 13:10:58,673]   DEBUG [ref] CompactedDBG::build(): Estimated number of minimizer occurring at least once: 35636198\n",
      "[2023-07-05 13:11:00,877]   DEBUG [ref] CompactedDBG::filter(): Processed 296860416 k-mers in 119563 reads\n",
      "[2023-07-05 13:11:00,877]   DEBUG [ref] CompactedDBG::filter(): Found 145970754 unique k-mers\n",
      "[2023-07-05 13:11:00,877]   DEBUG [ref] CompactedDBG::filter(): Number of blocks in Bloom filter is 1000243\n",
      "[2023-07-05 13:11:02,280]   DEBUG [ref] CompactedDBG::construct(): Extract approximate unitigs (1/2)\n",
      "[2023-07-05 13:11:36,552]   DEBUG [ref] CompactedDBG::construct(): Extract approximate unitigs (2/2)\n",
      "[2023-07-05 13:11:38,757]   DEBUG [ref] CompactedDBG::construct(): Closed all input files\n",
      "[2023-07-05 13:11:38,757]   DEBUG [ref] \n",
      "[2023-07-05 13:11:38,757]   DEBUG [ref] CompactedDBG::construct(): Splitting unitigs (1/2)\n",
      "[2023-07-05 13:11:39,158]   DEBUG [ref] \n",
      "[2023-07-05 13:11:39,158]   DEBUG [ref] CompactedDBG::construct(): Splitting unitigs (2/2)\n",
      "[2023-07-05 13:11:41,362]   DEBUG [ref] CompactedDBG::construct(): Before split: 1330136 unitigs\n",
      "[2023-07-05 13:11:41,362]   DEBUG [ref] CompactedDBG::construct(): After split (1/1): 1330136 unitigs\n",
      "[2023-07-05 13:11:41,362]   DEBUG [ref] CompactedDBG::construct(): Unitigs split: 1960\n",
      "[2023-07-05 13:11:41,362]   DEBUG [ref] CompactedDBG::construct(): Unitigs deleted: 0\n",
      "[2023-07-05 13:11:41,362]   DEBUG [ref] \n",
      "[2023-07-05 13:11:41,362]   DEBUG [ref] CompactedDBG::construct(): Joining unitigs\n",
      "[2023-07-05 13:11:58,297]   DEBUG [ref] CompactedDBG::construct(): After join: 1251507 unitigs\n",
      "[2023-07-05 13:11:58,297]   DEBUG [ref] CompactedDBG::construct(): Joined 79136 unitigs\n",
      "[2023-07-05 13:11:58,598]   DEBUG [ref] [build] extracting distinguishing flanking k-mers from \"/home/laura/projects/virus-watch-data/canine_ref/Canis_lupus_familiaris.ROS_Cfam_1.0.dna.toplevel.fa.gz\" \"/home/laura/projects/virus-watch-data/rhesus_ref/Macaca_mulatta.Mmul_10.dna.toplevel.fa.gz\"\n",
      "[2023-07-05 13:39:42,844]   DEBUG [ref] [build] identified 7948912 distinguishing flanking k-mers\n",
      "[2023-07-05 13:39:54,069]   DEBUG [ref] [build] building MPHF\n",
      "[2023-07-05 13:40:24,535]   DEBUG [ref] [build] creating equivalence classes ...\n",
      "[2023-07-05 13:41:17,554]   DEBUG [ref] [build] target de Bruijn graph has k-mer length 31 and minimizer length 23\n",
      "[2023-07-05 13:41:17,654]   DEBUG [ref] [build] target de Bruijn graph has 9835717 contigs and contains 151974779 k-mers\n",
      "[2023-07-05 13:41:53,727]   DEBUG [ref] \n",
      "[2023-07-05 13:41:55,893]   DEBUG [main] Removing `tmp` directory\n"
     ]
    }
   ],
   "source": [
    "!kb ref \\\n",
    "    --overwrite --verbose \\\n",
    "    -t 30 \\\n",
    "    -i $index \\\n",
    "    -g $t2g \\\n",
    "    -f1 $fa \\\n",
    "    --kallisto $kallisto \\\n",
    "    --bustools $bustools \\\n",
    "    $canine_fasta,$host_fasta $canine_gtf,$host_gtf"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "16b60cc4-c5a3-4e3e-b973-0e27ed7efb22",
   "metadata": {},
   "source": [
    "### Align to combined macaque and canine genomes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f3666977-d63e-4337-b565-2d9d2345fd07",
   "metadata": {},
   "outputs": [],
   "source": [
    "host_out_folder = f\"{fastq_folder}/../full_index/host_canine\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "81222734-7557-4e0d-abec-4e403388065e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[bus] will try running read files supplied in batch file\n",
      "[bus] Note: Strand option was not specified; setting it to --unstranded for specified technology\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 119,563\n",
      "[index] number of k-mers: 151,974,779\n",
      "[index] number of distinguishing flanking k-mers: 7,948,912\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698516_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698516_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698524_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698524_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698536_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698536_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698560_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698560_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698542_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698542_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698571_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698571_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698569_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698569_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698518_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698518_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698541_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698541_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698556_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698556_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698573_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698573_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698520_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698520_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698507_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698507_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698528_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698528_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698581_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698581_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698501_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698501_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698523_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698523_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698530_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698530_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698583_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698583_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698500_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698500_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698601_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698601_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698592_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698592_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698519_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698519_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698510_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698510_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698544_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698544_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698551_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698551_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698555_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698555_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698546_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698546_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698570_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698570_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698499_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698499_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698563_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698563_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698568_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698568_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698593_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698593_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698603_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698603_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698515_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698515_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698561_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698561_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698545_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698545_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698549_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698549_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698538_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698538_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698554_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698554_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698565_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698565_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698535_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698535_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698589_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698589_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698578_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698578_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698588_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698588_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698596_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698596_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698599_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698599_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698584_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698584_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698512_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698512_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698511_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698511_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698527_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698527_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698547_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698547_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698594_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698594_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698566_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698566_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698532_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698532_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698526_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698526_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698574_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698574_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698600_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698600_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698567_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698567_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698529_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698529_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698572_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698572_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698534_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698534_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698587_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698587_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698598_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698598_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698540_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698540_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698525_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698525_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698509_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698509_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698602_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698602_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698505_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698505_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698508_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698508_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698522_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698522_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698597_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698597_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698595_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698595_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698521_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698521_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698562_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698562_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698548_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698548_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698576_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698576_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698543_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698543_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698591_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698591_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698557_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698557_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698517_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698517_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698586_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698586_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698579_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698579_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698531_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698531_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698552_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698552_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698585_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698585_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698558_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698558_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698559_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698559_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698503_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698503_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698564_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698564_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698537_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698537_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698590_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698590_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698604_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698604_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698580_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698580_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698533_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698533_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698513_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698513_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698502_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698502_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698504_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698504_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698575_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698575_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698506_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698506_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698577_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698577_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698582_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698582_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698514_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698514_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698553_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698553_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698550_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698550_2.fastq.gz\n",
      "[progress] 30593M reads processed (28.9% mapped)             % mapped)             28.6% mapped)             28.7% mapped)             28.7% mapped)             % mapped)             28.7% mapped)             % mapped)             % mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             % mapped)             % mapped)             % mapped)             29.1% mapped)             29.2% mapped)             % mapped)             % mapped)             29.4% mapped)             29.5% mapped)             29.5% mapped)             29.4% mapped)             29.4% mapped)              (29.5% mapped)             29.5% mapped)             29.5% mapped)             29.5% mapped)             29.3% mapped)             29.2% mapped)             29.0% mapped)             29.0% mapped)             28.9% mapped)             28.8% mapped)             % mapped)             28.8% mapped)             % mapped)             28.7% mapped)             % mapped)             28.7% mapped)             28.7% mapped)             28.6% mapped)             28.6% mapped)             28.3% mapped)             28.3% mapped)             28.3% mapped)             % mapped)             28.2% mapped)             28.2% mapped)             28.1% mapped)             28.1% mapped)             28.1% mapped)             % mapped)             28.1% mapped)             28.1% mapped)             28.0% mapped)             % mapped)             27.9% mapped)             27.9% mapped)             27.9% mapped)             27.9% mapped)             % mapped)             27.8% mapped)             27.8% mapped)             27.8% mapped)             27.8% mapped)             27.8% mapped)             27.8% mapped)             27.7% mapped)             27.7% mapped)             27.7% mapped)             27.6% mapped)             27.6% mapped)             27.6% mapped)             27.6% mapped)             27.6% mapped)             % mapped)             27.6% mapped)             27.6% mapped)             27.6% mapped)             % mapped)             27.5% mapped)             27.5% mapped)             27.5% mapped)             % mapped)             27.5% mapped)             27.5% mapped)             % mapped)             27.5% mapped)             27.5% mapped)             27.5% mapped)             27.5% mapped)             27.5% mapped)             27.5% mapped)             27.4% mapped)             % mapped)             27.2% mapped)             27.2% mapped)             27.2% mapped)             27.2% mapped)             27.1% mapped)             % mapped)             % mapped)             27.0% mapped)             27.0% mapped)             27.0% mapped)             27.0% mapped)             27.0% mapped)             % mapped)             27.0% mapped)             27.0% mapped)             27.0% mapped)             % mapped)             27.0% mapped)             27.0% mapped)             27.0% mapped)             26.9% mapped)             26.9% mapped)             26.9% mapped)             26.9% mapped)             26.9% mapped)             26.9% mapped)             % mapped)             % mapped)             26.9% mapped)             26.9% mapped)             27.0% mapped)             26.9% mapped)             26.9% mapped)             26.9% mapped)             % mapped)             26.9% mapped)             26.9% mapped)             % mapped)             26.9% mapped)             27.0% mapped)             27.0% mapped)             27.0% mapped)             % mapped)             27.1% mapped)             % mapped)             27.1% mapped)             27.1% mapped)             27.2% mapped)             27.3% mapped)             27.3% mapped)             27.3% mapped)             27.3% mapped)             27.3% mapped)             27.3% mapped)             27.3% mapped)             27.4% mapped)             27.4% mapped)             % mapped)             27.4% mapped)             27.4% mapped)             27.5% mapped)             27.5% mapped)             27.5% mapped)             27.5% mapped)             27.5% mapped)             27.5% mapped)             27.6% mapped)             27.6% mapped)             27.6% mapped)             27.6% mapped)             27.7% mapped)             27.7% mapped)             27.8% mapped)             % mapped)             27.8% mapped)             % mapped)             27.8% mapped)             27.8% mapped)             27.8% mapped)             % mapped)             27.8% mapped)             % mapped)             % mapped)             27.9% mapped)             % mapped)             27.9% mapped)             % mapped)             27.9% mapped)             % mapped)             28.0% mapped)             28.0% mapped)             28.0% mapped)             28.0% mapped)             % mapped)             28.0% mapped)             28.1% mapped)             28.1% mapped)             % mapped)             % mapped)             28.1% mapped)             28.2% mapped)             28.3% mapped)             28.3% mapped)             28.3% mapped)             28.3% mapped)             % mapped)             28.4% mapped)             28.4% mapped)             28.4% mapped)             28.4% mapped)             28.4% mapped)             28.4% mapped)             28.4% mapped)             28.4% mapped)             28.4% mapped)             28.5% mapped)             28.5% mapped)             28.5% mapped)             % mapped)             28.5% mapped)             % mapped)             28.5% mapped)             % mapped)             28.6% mapped)             28.6% mapped)             28.6% mapped)             28.6% mapped)             28.6% mapped)             % mapped)             28.6% mapped)             28.6% mapped)             28.6% mapped)             28.6% mapped)             28.6% mapped)             28.6% mapped)             28.6% mapped)             % mapped)             28.6% mapped)             % mapped)             28.7% mapped)             % mapped)             28.7% mapped)             % mapped)             % mapped)             28.7% mapped)             28.7% mapped)             28.7% mapped)             28.8% mapped)             % mapped)             28.8% mapped)             % mapped)             28.8% mapped)             % mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)              mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             % mapped)             28.9% mapped)             % mapped)             28.9% mapped)             % mapped)             28.9% mapped)             % mapped)             % mapped)             28.9% mapped)             28.9% mapped)             28.9% mapped)             % mapped)             % mapped)             28.9% mapped)             28.9% mapped)             % mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             % mapped)             28.8% mapped)             28.8% mapped)             % mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.8% mapped)             28.9% mapped)             % mapped)             28.9% mapped)             28.9% mapped)              done\n",
      "[quant] processed 30,594,130,037 reads, 8,828,381,950 reads pseudoaligned\n",
      "\n",
      "\tCommand being timed: \"/home/laura/projects/kallisto/build/src/kallisto bus -i /home/laura/projects/virus-watch-data/canine_macaque_ref/index.idx -o /home/laura/projects/virus-watch-data/PRJNA665227/raw/../full_index/host_canine -t 30 -B /home/laura/projects/virus-watch-data/PRJNA665227/raw/batch.txt --batch-barcodes -x 0,0,12:0,12,20:1,0,0\"\n",
      "\tUser time (seconds): 917554.91\n",
      "\tSystem time (seconds): 8742.42\n",
      "\tPercent of CPU this job got: 2750%\n",
      "\tElapsed (wall clock) time (h:mm:ss or m:ss): 9:21:20\n",
      "\tAverage shared text size (kbytes): 0\n",
      "\tAverage unshared data size (kbytes): 0\n",
      "\tAverage stack size (kbytes): 0\n",
      "\tAverage total size (kbytes): 0\n",
      "\tMaximum resident set size (kbytes): 4760208\n",
      "\tAverage resident set size (kbytes): 0\n",
      "\tMajor (requiring I/O) page faults: 10\n",
      "\tMinor (reclaiming a frame) page faults: 4385010924\n",
      "\tVoluntary context switches: 368961\n",
      "\tInvoluntary context switches: 1924263\n",
      "\tSwaps: 0\n",
      "\tFile system inputs: 3555245832\n",
      "\tFile system outputs: 553203352\n",
      "\tSocket messages sent: 0\n",
      "\tSocket messages received: 0\n",
      "\tSignals delivered: 0\n",
      "\tPage size (bytes): 4096\n",
      "\tExit status: 0\n",
      "CPU times: user 11min 29s, sys: 2min 51s, total: 14min 21s\n",
      "Wall time: 9h 21min 20s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!/usr/bin/time -v $kallisto bus \\\n",
    "        -i $index \\\n",
    "        -o $host_out_folder \\\n",
    "        -t 30 \\\n",
    "        -B $sample_batch_file \\\n",
    "        --batch-barcodes \\\n",
    "        -x 0,0,12:0,12,20:1,0,0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d72e6b9d-5987-422e-9d3f-5400bbea88e4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "partition time: 3.62s\n",
      "partition time: 3.2s\n",
      "partition time: 2.94s\n",
      "partition time: 2.84s\n",
      "partition time: 2.67s\n",
      "partition time: 2.86s\n",
      "partition time: 2.78s\n",
      "partition time: 2.77s\n",
      "partition time: 2.79s\n",
      "partition time: 2.76s\n",
      "partition time: 2.78s\n",
      "partition time: 2.9s\n",
      "partition time: 2.91s\n",
      "partition time: 2.95s\n",
      "partition time: 3.15s\n",
      "partition time: 3.24s\n",
      "partition time: 3.33s\n",
      "partition time: 3.59s\n",
      "partition time: 4.04s\n",
      "partition time: 3.01s\n",
      "partition time: 2.81s\n",
      "partition time: 2.83s\n",
      "partition time: 2.72s\n",
      "partition time: 2.8s\n",
      "partition time: 2.84s\n",
      "partition time: 2.72s\n",
      "partition time: 2.82s\n",
      "partition time: 2.81s\n",
      "partition time: 2.88s\n",
      "partition time: 3.54s\n",
      "partition time: 3.74s\n",
      "partition time: 3.06s\n",
      "partition time: 3.14s\n",
      "partition time: 4.51s\n",
      "partition time: 3.84s\n",
      "partition time: 3.48s\n",
      "partition time: 2.98s\n",
      "partition time: 2.84s\n",
      "partition time: 2.71s\n",
      "partition time: 2.77s\n",
      "partition time: 2.83s\n",
      "partition time: 2.78s\n",
      "partition time: 2.83s\n",
      "partition time: 2.73s\n",
      "partition time: 2.79s\n",
      "partition time: 2.75s\n",
      "partition time: 2.74s\n",
      "partition time: 2.77s\n",
      "partition time: 2.82s\n",
      "partition time: 2.93s\n",
      "partition time: 2.91s\n",
      "partition time: 3.11s\n",
      "partition time: 3.2s\n",
      "partition time: 3.58s\n",
      "partition time: 3.64s\n",
      "partition time: 2.98s\n",
      "partition time: 3.04s\n",
      "partition time: 3.09s\n",
      "partition time: 3.11s\n",
      "partition time: 3.01s\n",
      "partition time: 2.99s\n",
      "partition time: 3.15s\n",
      "partition time: 3.12s\n",
      "partition time: 3.06s\n",
      "partition time: 3.38s\n",
      "partition time: 3.27s\n",
      "Read in 8828381950 BUS records\n",
      "reading time 232.6s\n",
      "sorting time 2258.33s\n",
      "writing time 0s\n",
      "CPU times: user 1min 1s, sys: 13.8 s, total: 1min 15s\n",
      "Wall time: 58min 49s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!$bustools sort \\\n",
    "    -m 4G \\\n",
    "    -t 30 \\\n",
    "    -o $host_out_folder/output_sorted.bus \\\n",
    "    $host_out_folder/output.bus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6843eafe-426f-400f-a17e-408fdb8399df",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Read in 1917532843 BUS records, wrote 801033 barcodes to whitelist with threshold 1000\n",
      "CPU times: user 252 ms, sys: 45.9 ms, total: 298 ms\n",
      "Wall time: 14.6 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# Generate onlist from data since SeqWell does not have a whitelist\n",
    "# threshold: Minimum number of times a barcode must appear to be included in onlist\n",
    "!$bustools whitelist \\\n",
    "    --threshold 1000 \\\n",
    "    -o $host_out_folder/bustools_whitelist.txt \\\n",
    "    $host_out_folder/output_sorted.bus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4749cde7-c136-4243-a1cd-d155dbd14df7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 775516 barcodes in the on-list\n",
      "Processed 1917532843 BUS records\n",
      "In on-list = 1236574388\n",
      "Corrected    = 181799192\n",
      "Uncorrected  = 499159263\n",
      "CPU times: user 9.77 s, sys: 1.95 s, total: 11.7 s\n",
      "Wall time: 9min 59s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!$bustools correct \\\n",
    "    -w $host_out_folder/bustools_whitelist.txt \\\n",
    "    -o $host_out_folder/output_sorted_corrected.bus \\\n",
    "    $host_out_folder/output_sorted.bus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "11a795cb-75a5-43ae-b7a0-1b2cf02e3c00",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "partition time: 2.34s\n",
      "partition time: 2.26s\n",
      "partition time: 2.17s\n",
      "partition time: 2.17s\n",
      "partition time: 2.08s\n",
      "partition time: 2.1s\n",
      "partition time: 2.03s\n",
      "partition time: 2.08s\n",
      "partition time: 2.11s\n",
      "partition time: 2.13s\n",
      "partition time: 1.16s\n",
      "Read in 1418373580 BUS records\n",
      "reading time 12.75s\n",
      "sorting time 291.5s\n",
      "writing time 0s\n",
      "CPU times: user 5.1 s, sys: 1.19 s, total: 6.29 s\n",
      "Wall time: 4min 54s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!$bustools sort \\\n",
    "    -m 4G \\\n",
    "    -t 30 \\\n",
    "    -o $host_out_folder/output_sorted_corrected_sorted.bus \\\n",
    "    $host_out_folder/output_sorted_corrected.bus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c9ffd455-005d-4ff4-9406-9c913652d6fe",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 4.67 s, sys: 1.01 s, total: 5.68 s\n",
      "Wall time: 4min 47s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!$bustools count \\\n",
    "    --genecounts \\\n",
    "    -o $host_out_folder/bustools_count/ \\\n",
    "    -g $t2g \\\n",
    "    -e $host_out_folder/matrix.ec \\\n",
    "    -t $host_out_folder/transcripts.txt \\\n",
    "    $host_out_folder/output_sorted_corrected_sorted.bus"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "13d0336f-1735-431b-826b-74468bed273d",
   "metadata": {},
   "source": [
    "### Align to PalmDB"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d98bf09c-aae4-4e07-915c-a1f8cfeff7f2",
   "metadata": {},
   "source": [
    "Concatenate macaque and canine genome files for d-list:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "f3f41bf6-47a6-4eb3-8f36-60ccf0f2b0a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat $canine_fasta $host_fasta > $canine_macaque_fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ab61263-64f3-4a03-b909-e8d4a990cfa4",
   "metadata": {},
   "source": [
    "Generate virus index:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "db743ece-2f41-4b6d-b828-b6aaef4d03a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "virus_fasta = \"/home/laura/projects/virus-watch-data/virus_ref/uniques_noduplicates.fa\"\n",
    "virus_index = \"/home/laura/projects/virus-watch-data/virus_ref/kallisto_aa/noduplicates_rhesus_canine.idx\"\n",
    "virus_t2g = \"/home/laura/projects/virus-watch-data/virus_ref/nodup_clu_t2g.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "227af422-8b38-4c62-9835-d11c5682c715",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[build] loading fasta file /home/laura/projects/virus-watch-data/virus_ref/uniques_noduplicates.fa\n",
      "[build] k-mer length: 31\n",
      "KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
      "KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
      "KmerStream::KmerStream(): Finished\n",
      "CompactedDBG::build(): Estimated number of k-mers occurring at least once: 37641510\n",
      "CompactedDBG::build(): Estimated number of minimizer occurring at least once: 7877811\n",
      "CompactedDBG::filter(): Processed 87630084 k-mers in 296561 reads\n",
      "CompactedDBG::filter(): Found 37509161 unique k-mers\n",
      "CompactedDBG::filter(): Number of blocks in Bloom filter is 257317\n",
      "CompactedDBG::construct(): Extract approximate unitigs (1/2)\n",
      "CompactedDBG::construct(): Extract approximate unitigs (2/2)\n",
      "CompactedDBG::construct(): Closed all input files\n",
      "\n",
      "CompactedDBG::construct(): Splitting unitigs (1/2)\n",
      "\n",
      "CompactedDBG::construct(): Splitting unitigs (2/2)\n",
      "CompactedDBG::construct(): Before split: 2040617 unitigs\n",
      "CompactedDBG::construct(): After split (1/1): 2040617 unitigs\n",
      "CompactedDBG::construct(): Unitigs split: 1554\n",
      "CompactedDBG::construct(): Unitigs deleted: 0\n",
      "\n",
      "CompactedDBG::construct(): Joining unitigs\n",
      "CompactedDBG::construct(): After join: 2021701 unitigs\n",
      "CompactedDBG::construct(): Joined 19270 unitigs\n",
      "[build] extracting distinguishing flanking k-mers from \"/home/laura/projects/virus-watch-data/canine_macaque_ref/combined_genomes.dna.toplevel.fa.gz\"\n",
      "[build] identified 90302 distinguishing flanking k-mers\n",
      "[build] building MPHF\n",
      "[build] creating equivalence classes ... \n",
      "[build] target de Bruijn graph has k-mer length 31 and minimizer length 23\n",
      "[build] target de Bruijn graph has 2081798 contigs and contains 37632058 k-mers \n",
      "\n",
      "\tCommand being timed: \"/home/laura/projects/kallisto/build/src/kallisto index -t 30 --aa --d-list=/home/laura/projects/virus-watch-data/canine_macaque_ref/combined_genomes.dna.toplevel.fa.gz -i /home/laura/projects/virus-watch-data/virus_ref/kallisto_aa/noduplicates_rhesus_canine.idx /home/laura/projects/virus-watch-data/virus_ref/uniques_noduplicates.fa\"\n",
      "\tUser time (seconds): 9472.81\n",
      "\tSystem time (seconds): 194.76\n",
      "\tPercent of CPU this job got: 499%\n",
      "\tElapsed (wall clock) time (h:mm:ss or m:ss): 32:16.66\n",
      "\tAverage shared text size (kbytes): 0\n",
      "\tAverage unshared data size (kbytes): 0\n",
      "\tAverage stack size (kbytes): 0\n",
      "\tAverage total size (kbytes): 0\n",
      "\tMaximum resident set size (kbytes): 4917780\n",
      "\tAverage resident set size (kbytes): 0\n",
      "\tMajor (requiring I/O) page faults: 0\n",
      "\tMinor (reclaiming a frame) page faults: 82118369\n",
      "\tVoluntary context switches: 102741\n",
      "\tInvoluntary context switches: 11266\n",
      "\tSwaps: 0\n",
      "\tFile system inputs: 48\n",
      "\tFile system outputs: 64348208\n",
      "\tSocket messages sent: 0\n",
      "\tSocket messages received: 0\n",
      "\tSignals delivered: 0\n",
      "\tPage size (bytes): 4096\n",
      "\tExit status: 0\n"
     ]
    }
   ],
   "source": [
    "# Generate virus reference index (with default d-list-overhang = 3)\n",
    "!/usr/bin/time -v $kallisto index \\\n",
    "    -t 30 \\\n",
    "    --aa \\\n",
    "    --d-list=$canine_macaque_fasta \\\n",
    "    -i $virus_index \\\n",
    "    $virus_fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7faf8797-a842-4254-8741-8f7923223563",
   "metadata": {},
   "source": [
    "Align to PalmDB and correct barcodes using host onlist:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d834169c-5dda-486a-8ae5-b29572d7f9d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_folder = f\"{fastq_folder}/../full_index/virus\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ba0e4621-fd11-4ef3-aaa5-84fddd61c7ff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/laura/projects/virus-watch-data/PRJNA665227/raw/batch.txt'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_batch_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "1be8eee3-0a2f-4b12-8454-7f150ad560a1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[bus] will try running read files supplied in batch file\n",
      "[bus] Note: Strand option was not specified; setting it to --unstranded for specified technology\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,632,058\n",
      "[index] number of distinguishing flanking k-mers: 90,302\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698516_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698516_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698524_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698524_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698536_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698536_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698560_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698560_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698542_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698542_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698571_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698571_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698569_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698569_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698518_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698518_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698541_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698541_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698556_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698556_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698573_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698573_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698520_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698520_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698507_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698507_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698528_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698528_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698581_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698581_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698501_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698501_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698523_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698523_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698530_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698530_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698583_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698583_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698500_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698500_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698601_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698601_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698592_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698592_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698519_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698519_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698510_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698510_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698544_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698544_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698551_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698551_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698555_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698555_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698546_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698546_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698570_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698570_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698499_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698499_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698563_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698563_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698568_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698568_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698593_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698593_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698603_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698603_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698515_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698515_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698561_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698561_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698545_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698545_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698549_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698549_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698538_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698538_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698554_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698554_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698565_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698565_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698535_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698535_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698589_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698589_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698578_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698578_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698588_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698588_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698596_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698596_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698599_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698599_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698584_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698584_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698512_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698512_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698511_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698511_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698527_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698527_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698547_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698547_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698594_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698594_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698566_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698566_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698532_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698532_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698526_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698526_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698574_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698574_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698600_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698600_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698567_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698567_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698529_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698529_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698572_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698572_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698534_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698534_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698587_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698587_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698598_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698598_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698540_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698540_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698525_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698525_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698509_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698509_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698602_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698602_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698505_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698505_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698508_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698508_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698522_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698522_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698597_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698597_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698595_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698595_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698521_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698521_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698562_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698562_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698548_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698548_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698576_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698576_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698543_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698543_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698591_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698591_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698557_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698557_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698517_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698517_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698586_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698586_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698579_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698579_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698531_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698531_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698552_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698552_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698585_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698585_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698558_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698558_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698559_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698559_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698503_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698503_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698564_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698564_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698537_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698537_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698590_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698590_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698604_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698604_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698580_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698580_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698533_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698533_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698513_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698513_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698502_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698502_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698504_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698504_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698575_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698575_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698506_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698506_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698577_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698577_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698582_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698582_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698514_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698514_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698553_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698553_2.fastq.gz\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698550_1.fastq.gz\n",
      "[quant] will process file 2: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698550_2.fastq.gz\n",
      "[progress] 2900M reads processed (0.7% mapped)             .7% mapped)             0.7% mapped)             0.7% mapped)             0.7% mapped)             0.7% mapped)             0.7% mapped)             0.7% mapped)             0.7% mapped)             0.7% mapped)             0.7% mapped)             CPU times: user 3min 33s, sys: 48.7 s, total: 4min 21s\n",
      "Wall time: 3h 18min 40s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!/usr/bin/time -v $kallisto bus \\\n",
    "        -i $virus_index \\\n",
    "        -o $out_folder \\\n",
    "        --aa \\\n",
    "        -t 30 \\\n",
    "        -B $sample_batch_file \\\n",
    "        --batch-barcodes \\\n",
    "        -x 0,0,12:0,12,20:1,0,0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "49ebbc06-8654-4864-93a6-4b8345a23d84",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "^C\n",
      "/bin/bash: correct: command not found\n",
      "Error: File not found, /home/laura/projects/virus-watch-data/PRJNA665227/raw/../full_index/virus/output_sorted_corrected.bus\n",
      "Usage: bustools sort [options] bus-files\n",
      "\n",
      "Options: \n",
      "Default behavior (with no flag) is to sort by barcode, UMI, ec, then flag\n",
      "-t, --threads         Number of threads to use\n",
      "-m, --memory          Maximum memory used\n",
      "-T, --temp            Location and prefix for temporary files \n",
      "                      required if using -p, otherwise defaults to output\n",
      "-o, --output          File for sorted output\n",
      "-p, --pipe            Write to standard output\n",
      "    --umi             Sort by UMI, barcode, then ec\n",
      "    --count           Sort by multiplicity, barcode, UMI, then ec\n",
      "    --flags           Sort by flag, ec, barcode, then UMI\n",
      "    --flags-bc        Sort by flag, barcode, UMI, then ec\n",
      "\n",
      "Error: File not found /home/laura/projects/virus-watch-data/PRJNA665227/raw/../full_index/virus/matrix.ec\n",
      "Error: File not found /home/laura/projects/virus-watch-data/PRJNA665227/raw/../full_index/virus/transcripts.txt\n",
      "Usage: bustools count [options] sorted-bus-files\n",
      "\n",
      "Options: \n",
      "-o, --output          Output directory gene matrix files\n",
      "-g, --genemap         File for mapping transcripts to genes\n",
      "-e, --ecmap           File for mapping equivalence classes to transcripts\n",
      "-t, --txnames         File with names of transcripts\n",
      "    --genecounts      Aggregate counts to genes only\n",
      "    --umi-gene        Perform gene-level collapsing of UMIs\n",
      "    --em              Estimate gene abundances using EM algorithm\n",
      "    --cm              Count multiplicites instead of UMIs\n",
      "-s, --split           Split output matrix in two (plus ambiguous) based on transcripts supplied in this file\n",
      "-m, --multimapping    Include bus records that pseudoalign to multiple genes\n",
      "    --hist            Output copy per UMI histograms for all genes\n",
      "-d  --downsample      Specify a factor between 0 and 1 specifying how much to downsample\n",
      "    --rawcounts       The count matrix will contain raw counts instead of UMI counts\n",
      "\n",
      "CPU times: user 12.7 ms, sys: 27.5 ms, total: 40.3 ms\n",
      "Wall time: 732 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "!$bustools sort \\\n",
    "    -m 4G \\\n",
    "    -t 20 \\\n",
    "    -o $out_folder/output_sorted.bus \\\n",
    "    $out_folder/output.bus\n",
    "\n",
    "!$bustools correct \\\n",
    "    -w $host_out_folder/bustools_whitelist.txt \\\n",
    "    -o $out_folder/output_sorted_corrected.bus \\\n",
    "    $out_folder/output_sorted.bus\n",
    "\n",
    "!$bustools sort \\\n",
    "    -m 4G \\\n",
    "    -t 30 \\\n",
    "    -o $out_folder/output_sorted_corrected_sorted.bus \\\n",
    "    $out_folder/output_sorted_corrected.bus\n",
    "\n",
    "!$bustools count \\\n",
    "    --genecounts \\\n",
    "    -o $out_folder/bustools_count/ \\\n",
    "    -g $virus_t2g \\\n",
    "    -e $out_folder/matrix.ec \\\n",
    "    -t $out_folder/transcripts.txt \\\n",
    "    $out_folder/output_sorted_corrected_sorted.bus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e091564a-7b0e-4cce-80e0-b0ea4aeac218",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
