{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "210dc559-9078-463f-9ea4-92fdcf5a4aa7",
   "metadata": {},
   "source": [
    "# Create Kronaplot for all viruses across all cells grouped by animal and timepoint:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "4b483aff-0760-46e2-be71-d5c8a1c8b1c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "bp = \"PRJNA665227\"\n",
    "fastq_folder = f\"/home/laura/projects/virus-watch-data/{bp}\"\n",
    "u_tax = f\"/home/laura/projects/virus-watch-data/virus_ref/u_tax_nodup_clu.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "6f314d09-f98e-4a02-880e-391976195296",
   "metadata": {},
   "outputs": [],
   "source": [
    "import anndata\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "def nd(arr):\n",
    "    \"\"\"\n",
    "    Function to transform numpy matrix to nd array.\n",
    "    \"\"\"\n",
    "    return np.asarray(arr).reshape(-1)\n",
    "\n",
    "# Path to Krona tool\n",
    "ktImportText = \"/home/laura/bin/ktImportText\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d59ddee-1037-4544-84a3-855927a851e2",
   "metadata": {},
   "source": [
    "___\n",
    "# Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "d11f8130-1b0d-4156-a788-2822c8ffc7d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/laura/anaconda3/lib/python3.9/site-packages/anndata/_core/anndata.py:1828: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.\n",
      "  utils.warn_names_duplicates(\"obs\")\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 225898 × 11176\n",
       "    obs: 'sample_barcode', 'srr', 'Run', 'donor_animal', 'Experiment', 'mdck_spike_in', 'hours_post_innoculation', 'day_post_infection', 'dpi', 'dpi_accessions', 'dpi_clean', 'leiden', 'celltype_clusters', 'celltype', 'species', 'dpi_clean_merged'\n",
       "    var: 'v_type'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "palmdb_adata = anndata.read(f\"{fastq_folder}/host_flag/virus_cdna_dna_masked_meta.h5ad\")\n",
    "palmdb_adata"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ba0fb21-df3d-4211-8305-14fc2b6f2540",
   "metadata": {},
   "source": [
    "Only keep macaque viruses and cells that passed QC:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "8c31df21-bddc-40ed-93d8-17d1e1696bd6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 195399 × 80\n",
       "    obs: 'sample_barcode', 'srr', 'Run', 'donor_animal', 'Experiment', 'mdck_spike_in', 'hours_post_innoculation', 'day_post_infection', 'dpi', 'dpi_accessions', 'dpi_clean', 'leiden', 'celltype_clusters', 'celltype', 'species', 'dpi_clean_merged'\n",
       "    var: 'v_type'"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "palmdb_adata = palmdb_adata[palmdb_adata.obs[\"celltype\"].notnull(), palmdb_adata.var[\"v_type\"] != \"below_threshold\"].copy()\n",
    "palmdb_adata"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "50215e1d-d35d-40ca-ac14-ea86039c6962",
   "metadata": {},
   "source": [
    "Load ID to phylogeny mapping:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "17c62507-5143-4a5a-9431-e076ea02b87b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rep_ID</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>u1</td>\n",
       "      <td>Pisuviricota</td>\n",
       "      <td>Pisoniviricetes</td>\n",
       "      <td>Nidovirales</td>\n",
       "      <td>Coronaviridae</td>\n",
       "      <td>Betacoronavirus</td>\n",
       "      <td>Severe acute respiratory syndrome-related coro...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>u10</td>\n",
       "      <td>Negarnaviricota</td>\n",
       "      <td>Monjiviricetes</td>\n",
       "      <td>Mononegavirales</td>\n",
       "      <td>Filoviridae</td>\n",
       "      <td>Ebolavirus</td>\n",
       "      <td>Zaire ebolavirus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>u100</td>\n",
       "      <td>Kitrinoviricota</td>\n",
       "      <td>Flasuviricetes</td>\n",
       "      <td>Amarillovirales</td>\n",
       "      <td>Flaviviridae</td>\n",
       "      <td>Flavivirus</td>\n",
       "      <td>West Nile virus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>u102</td>\n",
       "      <td>Kitrinoviricota</td>\n",
       "      <td>Alsuviricetes</td>\n",
       "      <td>Hepelivirales</td>\n",
       "      <td>Hepeviridae</td>\n",
       "      <td>Orthohepevirus</td>\n",
       "      <td>Orthohepevirus A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>909</th>\n",
       "      <td>u113</td>\n",
       "      <td>Negarnaviricota</td>\n",
       "      <td>Monjiviricetes</td>\n",
       "      <td>Mononegavirales</td>\n",
       "      <td>Paramyxoviridae</td>\n",
       "      <td>Morbillivirus</td>\n",
       "      <td>Measles morbillivirus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296556</th>\n",
       "      <td>u296608</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296557</th>\n",
       "      <td>u296609</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296558</th>\n",
       "      <td>u296613</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296559</th>\n",
       "      <td>u296616</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296560</th>\n",
       "      <td>u296623</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99228 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         rep_ID           phylum            class            order  \\\n",
       "0            u1     Pisuviricota  Pisoniviricetes      Nidovirales   \n",
       "409         u10  Negarnaviricota   Monjiviricetes  Mononegavirales   \n",
       "475        u100  Kitrinoviricota   Flasuviricetes  Amarillovirales   \n",
       "590        u102  Kitrinoviricota    Alsuviricetes    Hepelivirales   \n",
       "909        u113  Negarnaviricota   Monjiviricetes  Mononegavirales   \n",
       "...         ...              ...              ...              ...   \n",
       "296556  u296608                .                .                .   \n",
       "296557  u296609                .                .                .   \n",
       "296558  u296613                .                .                .   \n",
       "296559  u296616                .                .                .   \n",
       "296560  u296623                .                .                .   \n",
       "\n",
       "                 family            genus  \\\n",
       "0         Coronaviridae  Betacoronavirus   \n",
       "409         Filoviridae       Ebolavirus   \n",
       "475        Flaviviridae       Flavivirus   \n",
       "590         Hepeviridae   Orthohepevirus   \n",
       "909     Paramyxoviridae    Morbillivirus   \n",
       "...                 ...              ...   \n",
       "296556                .                .   \n",
       "296557                .                .   \n",
       "296558                .                .   \n",
       "296559                .                .   \n",
       "296560                .                .   \n",
       "\n",
       "                                                  species  \n",
       "0       Severe acute respiratory syndrome-related coro...  \n",
       "409                                      Zaire ebolavirus  \n",
       "475                                       West Nile virus  \n",
       "590                                      Orthohepevirus A  \n",
       "909                                 Measles morbillivirus  \n",
       "...                                                   ...  \n",
       "296556                                                  .  \n",
       "296557                                                  .  \n",
       "296558                                                  .  \n",
       "296559                                                  .  \n",
       "296560                                                  .  \n",
       "\n",
       "[99228 rows x 7 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id2tax = pd.read_csv(u_tax)\n",
    "\n",
    "# Drop columns not needed here and drop taxonomyduplicates\n",
    "id2tax = id2tax.drop(\"ID\", axis=1).drop(\"strandedness\", axis=1)\n",
    "id2tax = id2tax.drop_duplicates()\n",
    "\n",
    "id2tax"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0085c9f-550c-4bb7-b789-956f9b6765fb",
   "metadata": {},
   "source": [
    "___\n",
    "# Generate Krona html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "7a61fe4d-5301-4339-a978-dc24493304c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing krona.html...\n",
      "CPU times: user 4.05 s, sys: 0 ns, total: 4.05 s\n",
      "Wall time: 4.69 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "master = pd.DataFrame()\n",
    "for idx, timepoint in enumerate(palmdb_adata.obs[\"dpi_clean_merged\"].unique()):\n",
    "    for animal_id in palmdb_adata.obs[palmdb_adata.obs[\"dpi_clean_merged\"] == timepoint][\"donor_animal\"].unique():\n",
    "        adata_temp = palmdb_adata[(palmdb_adata.obs[\"dpi_clean_merged\"] == timepoint) & (palmdb_adata.obs[\"donor_animal\"] == animal_id), :]\n",
    "\n",
    "        # Add total number of counts (across all cells) for each virus ID to phylogeny data temp\n",
    "        virus_ids = adata_temp.var.index.values\n",
    "        total_counts = nd(adata_temp.X.sum(axis=0))\n",
    "        total_count_dict = {virus_ids[i]: total_counts[i] for i in range(len(virus_ids))}\n",
    "\n",
    "        phylogeny_data_temp = id2tax.copy()\n",
    "        phylogeny_data_temp['total_count'] = phylogeny_data_temp['rep_ID'].map(total_count_dict)\n",
    "        \n",
    "        # Drop viruses not in filter list\n",
    "        phylogeny_data_temp = phylogeny_data_temp.dropna()\n",
    "\n",
    "        # Remove non-relevent columns and change order of columns  \n",
    "        phylogeny_data_temp = phylogeny_data_temp[[\"total_count\", \"phylum\", \"class\", \"order\", \"family\", \"genus\", \"species\", \"rep_ID\"]]\n",
    "\n",
    "        # Replace dots with NaN\n",
    "        phylogeny_data_temp = phylogeny_data_temp.replace(\".\", np.nan)\n",
    "\n",
    "        # Add column with timepoint\n",
    "        phylogeny_data_temp[\"timepoint\"] = timepoint\n",
    "\n",
    "        # Add column with animal id\n",
    "        phylogeny_data_temp[\"animal_id\"] = animal_id\n",
    "\n",
    "        # Append to master dataframe\n",
    "        if idx == 0:\n",
    "            master = phylogeny_data_temp.copy()\n",
    "        else:\n",
    "            master = pd.concat([master, phylogeny_data_temp])\n",
    "\n",
    "# Save counts + taxnomomies data to txt\n",
    "master.to_csv(f'krona.txt', sep ='\\t', header=None, index=False)\n",
    "\n",
    "# Generate Krona plot\n",
    "krona_out = \"krona.html\"\n",
    "!$ktImportText krona.txt -o $krona_out -n \"Virus-positive cells\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "a0337605-50dd-4086-b35b-5a72358793be",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 149M\n",
      "-rw-rw-r-- 1 laura laura  160K Apr  3  2023 1_data_download.ipynb\n",
      "-rw-rw-r-- 1 laura laura   34K Oct 18 20:34 2.1_align_HostGenomeTranscriptome_noGTF.ipynb\n",
      "-rw-rw-r-- 1 laura laura  154K Oct 28 19:18 2.1_align_remove_reads_bus_AlignSRRindividually.ipynb\n",
      "-rw-rw-r-- 1 laura laura  287K Oct 31 07:34 2.1_align_remove_reads_bus+d-list.ipynb\n",
      "-rw-rw-r-- 1 laura laura  251K Oct 28 13:35 2.1_align_remove_reads_bus.ipynb\n",
      "-rw-rw-r-- 1 laura laura   26K Oct 27 14:43 2.1_align_remove_reads_bus_testing.ipynb\n",
      "-rw-rw-r-- 1 laura laura   50K Oct 18 07:59 2.1_align_virus_host_transcriptome+genome+ambigious.ipynb\n",
      "-rw-rw-r-- 1 laura laura   54K Oct 16 11:15 2.1_align_virus_host_transcriptome+genome_CheckEachKmer.ipynb\n",
      "-rw-rw-r-- 1 laura laura   11K Oct 18 16:45 2.1_align_virus_host_transcriptome+genome_DlistAll.ipynb\n",
      "-rw-rw-r-- 1 laura laura   43K Oct  7 11:42 2.1_align_virus_host_transcriptome+genome.ipynb\n",
      "-rw-rw-r-- 1 laura laura   48K Oct  4 16:37 2.1_align_virus_host_transcriptome.ipynb\n",
      "-rw-rw-r-- 1 laura laura   49K Oct 11 23:47 2.1_align_virus_no-host-mask.ipynb\n",
      "-rw-rw-r-- 1 laura laura   48K Oct  3 12:12 2.1_align_virus_overhang3.ipynb\n",
      "-rw-rw-r-- 1 laura laura   37K Aug 25 19:30 2.1_align_virus_to_bwa_host_removed.ipynb\n",
      "-rw-rw-r-- 1 laura laura   17K Oct 20 15:36 2.2_align_alphacoronavirus.ipynb\n",
      "-rw-rw-r-- 1 laura laura   42K Jul 17 00:47 2.2_align_dengue.ipynb\n",
      "-rw-rw-r-- 1 laura laura   40K Oct  8 23:07 2.2_align_enterovirusC.ipynb\n",
      "-rw-rw-r-- 1 laura laura   11K Aug 11 14:12 2.2_align_macdisease_viruses.ipynb\n",
      "-rw-rw-r-- 1 laura laura   45K Oct 19 16:01 2.2_align_OV_sars_cov2.ipynb\n",
      "-rw-rw-r-- 1 laura laura   46K Jul 14 02:30 2.2_align_SARSCoV.ipynb\n",
      "-rw-rw-r-- 1 laura laura   11K Aug  7 13:09 2.3_align_host_against_virus.ipynb\n",
      "-rw-rw-r-- 1 laura laura   88K Aug 15 18:00 2_align_host+canine+virus.ipynb\n",
      "-rw-rw-r-- 1 laura laura  931K Jul 15 16:28 3.1_host_QC.ipynb\n",
      "-rw-rw-r-- 1 laura laura  1.2M Jul 18 17:14 3.2_host_normalization_clustering.ipynb\n",
      "-rw-rw-r-- 1 laura laura  376K Jul 13 19:29 3.3_host_celltype_assignment.ipynb\n",
      "-rw-rw-r-- 1 laura laura  414K Oct 28 15:24 3.4_viral_QC_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  471K Oct 28 15:24 3.4_viral_QC_cdna+dna+amb.ipynb\n",
      "-rw-rw-r-- 1 laura laura  476K Oct 28 15:24 3.4_viral_QC_cdna+dna.ipynb\n",
      "-rw-rw-r-- 1 laura laura  501K Oct 28 15:24 3.4_viral_QC_cdna.ipynb\n",
      "-rw-rw-r-- 1 laura laura  476K Oct 28 15:24 3.4_viral_QC_CheckEachKmer.ipynb\n",
      "-rw-rw-r-- 1 laura laura  482K Oct 28 15:24 3.4_viral_QC.ipynb\n",
      "-rw-rw-r-- 1 laura laura  522K Oct 28 15:24 3.4_viral_QC_no-host-mask.ipynb\n",
      "-rw-rw-r-- 1 laura laura  651K Nov  6 16:07 3.4_viral_QC_remove-reads-bus+d-list.ipynb\n",
      "-rw-rw-r-- 1 laura laura  603K Oct 28 15:26 3.4_viral_QC_remove-reads-bus.ipynb\n",
      "-rw-rw-r-- 1 laura laura  728K Jul 24 16:39 3.5_host_celltype_fractions.ipynb\n",
      "-rw-rw-r-- 1 laura laura   15K Oct 31 10:17 3.6_bwa_vs_dlist.csv\n",
      "-rw-rw-r-- 1 laura laura  2.3M Nov  2 14:26 3.6_bwa_vs_dlist.ipynb\n",
      "-rw-rw-r-- 1 laura laura  2.5M Oct 19 16:07 3_OV_SARS_cov2_QC.ipynb\n",
      "-rw-rw-r-- 1 laura laura  586K Sep 19 03:05 4.1_qPCR_clinical_outcome_correlations_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  336K Oct 31 16:31 4.1_qPCR_clinical_outcome_correlations_cdna_dna_uncorrected.ipynb\n",
      "-rw-rw-r-- 1 laura laura  602K Sep 19 03:07 4.1_qPCR_clinical_outcome_correlations.ipynb\n",
      "-rw-rw-r-- 1 laura laura  1.3K Oct 31 18:13 4.2_fractions_vtype_per_tax_bus+d-list.csv\n",
      "-rw-rw-r-- 1 laura laura  1.2K Sep 19 03:50 4.2_fractions_vtype_per_tax_bwa.csv\n",
      "-rw-rw-r-- 1 laura laura  1.6K Sep 19 04:36 4.2_fractions_vtype_per_tax.csv\n",
      "-rw-rw-r-- 1 laura laura   299 Oct 31 18:16 4.2_ID2tax_top100_bus+d-list.csv\n",
      "-rw-rw-r-- 1 laura laura   368 Sep 20 05:07 4.2_ID2tax_top100_bwa.csv\n",
      "-rw-rw-r-- 1 laura laura   910 Oct  7 15:40 4.2_ID2tax_top100_cdna_dna.csv\n",
      "-rw-rw-r-- 1 laura laura  1.2K Sep 19 03:05 4.2_ID2tax_top100.csv\n",
      "-rw-rw-r-- 1 laura laura  3.2K Oct 31 18:17 4.2_ID2tax_top100_shared_bus+d-list.csv\n",
      "-rw-rw-r-- 1 laura laura  1.8K Sep 19 03:18 4.2_ID2tax_top100_shared_bwa.csv\n",
      "-rw-rw-r-- 1 laura laura  3.9K Sep 19 03:30 4.2_ID2tax_top100_shared.csv\n",
      "-rw-rw-r-- 1 laura laura  169K Aug 10 11:58 4.2_vir_fracs_celltype.csv\n",
      "-rw-rw-r-- 1 laura laura  173K Jul 15 14:05 4.2_vir_fracs_tp_animal_archive.csv\n",
      "-rw-rw-r-- 1 laura laura  3.2K Oct 31 18:12 4.2_vir_fracs_tp_animal_bus+d-list.csv\n",
      "-rw-rw-r-- 1 laura laura  3.9K Sep 20 05:07 4.2_vir_fracs_tp_animal_bwa.csv\n",
      "-rw-rw-r-- 1 laura laura   16K Oct  7 11:57 4.2_vir_fracs_tp_animal_cdna_dna.csv\n",
      "-rw-rw-r-- 1 laura laura   21K Sep 19 03:05 4.2_vir_fracs_tp_animal.csv\n",
      "-rw-rw-r-- 1 laura laura   47K Oct 31 18:13 4.2_vir_fracs_tp_animal_shared_bus+d-list.csv\n",
      "-rw-rw-r-- 1 laura laura   47K Oct 31 17:15 4.2_vir_fracs_tp_animal_shared_bwa.csv\n",
      "-rw-rw-r-- 1 laura laura   56K Sep 19 03:30 4.2_vir_fracs_tp_animal_shared.csv\n",
      "-rw-rw-r-- 1 laura laura  766K Nov  1 14:09 4.2_virus_fractions_bus+d-list.ipynb\n",
      "-rw-rw-r-- 1 laura laura  881K Sep 20 05:11 4.2_virus_fractions_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura 1005K Oct  7 17:23 4.2_virus_fractions_cdna_dna.ipynb\n",
      "-rw-rw-r-- 1 laura laura  1.2M Sep 19 04:36 4.2_virus_fractions.ipynb\n",
      "-rw-rw-r-- 1 laura laura  1.4M Oct 31 20:38 4.3_virus_per_celltypes_bus+d-list.ipynb\n",
      "-rw-rw-r-- 1 laura laura  1.1M Sep 20 07:32 4.3_virus_per_celltypes_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  1.4M Sep 19 03:06 4.3_virus_per_celltypes.ipynb\n",
      "-rw-rw-r-- 1 laura laura  3.0M Oct 31 21:10 4.4_common_macaque_infections_bus+d-list.ipynb\n",
      "-rw-rw-r-- 1 laura laura  3.0M Sep 19 03:07 4.4_common_macaque_infections_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  3.0M Sep 19 03:09 4.4_common_macaque_infections.ipynb\n",
      "-rw-rw-r-- 1 laura laura  425K Aug  7 12:29 4.5_virus_correlations.ipynb\n",
      "-rw-rw-r-- 1 laura laura  2.0M Nov  2 16:30 4.6_virus_profiles_+_host_expression_bus+d-list.ipynb\n",
      "-rw-rw-r-- 1 laura laura  967K Sep 20 09:03 4.6_virus_profiles_+_host_expression_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  700K Sep 19 04:05 4.6_virus_profiles_+_host_expression.ipynb\n",
      "-rw-rw-r-- 1 laura laura   27K Nov  8 10:53 4.7_krona_plot.ipynb\n",
      "-rw-rw-r-- 1 laura laura  4.6M Sep 19 00:35 4.8_virus_l1_distances_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  4.6M Oct  7 18:04 4.8_virus_l1_distances_cdna_dna.ipynb\n",
      "-rw-rw-r-- 1 laura laura  4.6M Sep 19 03:10 4.8_virus_l1_distances.ipynb\n",
      "-rw-rw-r-- 1 laura laura  168K Aug  1 12:22 5.1_linear_models.ipynb\n",
      "-rw-rw-r-- 1 laura laura  1.2M Jul 26 15:55 5_linear_models.ipynb\n",
      "-rw-rw-r-- 1 laura laura  725K Sep 19 20:56 6_pca_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  612K Sep 19 20:52 6_pca.ipynb\n",
      "-rw-rw-r-- 1 laura laura  461K Sep 19 03:12 7_deseq2_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  623K Sep 19 03:13 7_deseq2.ipynb\n",
      "drwxrwxr-x 3 laura laura  4.0K Aug 11 13:04 archive\n",
      "drwxrwxr-x 2 laura laura  4.0K Oct 19 13:51 blast_results\n",
      "-rw-rw-r-- 1 laura laura   51K Oct 25 16:05 bowtie2_coronavirus_alignment.ipynb\n",
      "-rw-rw-r-- 1 laura laura   30K Oct 25 17:22 bowtie2_virus+host_doubles_alignment.ipynb\n",
      "-rw-rw-r-- 1 laura laura  2.1K Apr 27  2023 clinical_scores_fixid.tsv\n",
      "-rw-rw-r-- 1 laura laura  4.8K Oct 19 12:10 extract_reads.py\n",
      "drwxrwxr-x 3 laura laura   20K Nov  7 18:01 figures\n",
      "-rw-rw-r-- 1 laura laura  1.6M Nov  7 18:21 hv_l2_donor_time_gene_weights.ipynb\n",
      "-rw-rw-r-- 1 laura laura  137K Nov  6 15:25 hv_l2_donor_time_gene_weights.xlsx\n",
      "-rw-rw-r-- 1 laura laura  472K Nov  8 10:54 krona.html\n",
      "-rw-rw-r-- 1 laura laura  133K Nov  8 10:54 krona.txt\n",
      "-rw-rw-r-- 1 laura laura   904 Jul  3 12:19 marker_list.csv\n",
      "-rw-rw-r-- 1 laura laura  183K Jul 25 20:53 norm_multilinear_top20_dict_lasso_immune.pickle\n",
      "-rw-rw-r-- 1 laura laura  8.8M Jul 24 14:29 norm_multilinear_top20_dict.pickle\n",
      "-rw-rw-r-- 1 laura laura   129 Mar 22  2023 notes.txt\n",
      "-rw-rw-r-- 1 laura laura   74K Mar 10  2023 PRJNA665227_data.json\n",
      "-rw-rw-r-- 1 laura laura   24K Sep 19 06:47 pull_out_reads_all_archive.csv\n",
      "-rw-rw-r-- 1 laura laura   35K Oct  3 15:10 pull_out_reads_all_bwa.csv\n",
      "-rw-rw-r-- 1 laura laura   46K Oct 10 16:12 pull_out_reads_all_cdna_dna.csv\n",
      "-rw-rw-r-- 1 laura laura   68K Sep 29 17:15 pull_out_reads_all_new.csv\n",
      "-rw-rw-r-- 1 laura laura  891K Sep 20 03:26 pull_out_reads_archive.ipynb\n",
      "-rw-rw-r-- 1 laura laura   57K Oct 20 12:18 pull_out_reads_blast+_cdna_dna.csv\n",
      "-rw-rw-r-- 1 laura laura   46M Nov  6 16:46 pull_out_reads_BLAST_comparisons_archive.ipynb\n",
      "-rw-rw-r-- 1 laura laura   16M Nov  6 16:50 pull_out_reads_BLAST_comparisons.ipynb\n",
      "-rw-rw-r-- 1 laura laura  7.2M Oct 23 11:02 pull_out_reads_blast+_nt_archive.csv\n",
      "-rw-rw-r-- 1 laura laura  1.5M Oct  3 15:11 pull_out_reads_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  1.2M Oct 10 16:50 pull_out_reads_cdna_dna.ipynb\n",
      "-rw-rw-r-- 1 laura laura  1.5M Oct  2 10:59 pull_out_reads.ipynb\n",
      "-rw-rw-r-- 1 laura laura   218 Sep 29 16:26 pull_out_reads_Kristjan.fa\n",
      "-rw-rw-r-- 1 laura laura   13K Oct  3 12:09 pull_out_reads_Kristjan.ipynb\n",
      "-rw-rw-r-- 1 laura laura  228K Oct 19 17:38 pull_out_reads_OV_sars_cov2.ipynb\n",
      "-rw-rw-r-- 1 laura laura     7 Nov  1 02:58 pull_out_reads_transcript_ids_temp.txt\n",
      "-rw-rw-r-- 1 laura laura  111K Aug 17 16:17 pull_out_reads_u131745.csv\n",
      "-rw-rw-r-- 1 laura laura  106K Aug 20 09:05 pull_out_reads_u131745_translated.csv\n",
      "-rw-rw-r-- 1 laura laura   13K Aug 18 12:46 pull_out_reads_u131745_translated.fasta\n",
      "-rw-rw-r-- 1 laura laura  4.0K Aug 17 16:19 pull_out_reads_u26830.csv\n",
      "-rw-rw-r-- 1 laura laura  1.9K Aug 17 18:45 pull_out_reads_u26830_translated.csv\n",
      "-rw-rw-r-- 1 laura laura  8.8M Jul 24 14:29 raw_multilinear_top20_dict.pickle\n",
      "-rw-rw-r-- 1 laura laura   95K Aug 24 21:04 remove_host_bwa.ipynb\n",
      "-rw-rw-r-- 1 laura laura  8.3K Aug 18 18:23 remove_host_DIAMOND.ipynb\n",
      "-rw-rw-r-- 1 laura laura   15K Jul 27 16:02 simpleRegression.py\n",
      "-rw-rw-r-- 1 laura laura   43K Mar 10  2023 SraRunTable.txt\n",
      "-rw-rw-r-- 1 laura laura   45K Oct  3 12:06 SRR12698505_2_extracted_u11842_new.fa\n",
      "-rw-rw-r-- 1 laura laura   65K Oct  3 12:06 SRR12698505_2_extracted_u34866_new.fa\n",
      "drwxrwxr-x 3 laura laura  4.0K Aug 11 13:54 stringent_dlist\n",
      "-rw-rw-r-- 1 laura laura  5.3K Nov  2 11:55 temp.csv\n",
      "-rw-rw-r-- 1 laura laura    90 Nov  2 11:53 temp.fa\n",
      "drwxrwxr-x 2 laura laura  4.0K Oct  3 11:16 test\n",
      "-rw-rw-r-- 1 laura laura  487K Aug 19 00:43 tmp.ipynb\n",
      "-rw-rw-r-- 1 laura laura  2.1K Jul 21 12:43 tombusviridae_comparison.fa\n",
      "-rw-rw-r-- 1 laura laura    10 Oct 19 13:54 txnames.txt\n",
      "-rw-rw-r-- 1 laura laura  2.0K Jul 18 19:11 UMI_overlap.ipynb\n",
      "-rw-rw-r-- 1 laura laura  7.4K Oct 19 17:57 Untitled1.ipynb\n",
      "-rw-rw-r-- 1 laura laura   17K Jul 25 12:42 Untitled.ipynb\n",
      "-rw-rw-r-- 1 laura laura   922 Apr 27  2023 viral_loads_fixid.tsv\n",
      "-rw-rw-r-- 1 laura laura   40K Aug 10 12:30 virus_dif_before_after_dlist_fix.ipynb\n",
      "-rw-rw-r-- 1 laura laura   60K Oct 31 16:50 virus_strandedness_bus+d-list.ipynb\n",
      "-rw-rw-r-- 1 laura laura  215K Jun 25 17:06 workflow_comparisons.ipynb\n"
     ]
    }
   ],
   "source": [
    "!ls -lh"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33dfd28b-5e5d-4009-8ef8-588dc042d0cb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
