{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "c482d079-8c0f-4bee-adff-85dd2c359b54",
   "metadata": {},
   "outputs": [],
   "source": [
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "59caaa49-a1f6-4b9d-988c-ffb94eba3eb3",
   "metadata": {},
   "outputs": [],
   "source": [
    "out = \"/home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9aacd273-113b-4d9d-9305-8080a1a25823",
   "metadata": {},
   "source": [
    "### Align using protein Kraken2 (translated search):"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0e51015d-535c-432f-9fa3-dab5454a92fc",
   "metadata": {},
   "source": [
    "Create ebov amino acid ref with NCBI taxID and align with Kraken translated search:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ad4d1a6c-714e-4078-bce4-a8d9c11437b1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading protein accession to taxon map... done.\n",
      "Downloaded accession to taxon map(s)\n",
      "Downloading taxonomy tree data... done.\n",
      "Uncompressing taxonomy data... done.\n",
      "Untarring taxonomy tree data... done.\n"
     ]
    }
   ],
   "source": [
    "!cd /home/laura/projects/kraken2-2.1.2 \\\n",
    "    && kraken2-build --db kraken_ebov_protein --protein --download-taxonomy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e966d788-d482-4c1e-a419-70c58c64bdfd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ebola_rdrp_aa.fasta contains the Ebola RdRP amino acid sequence from PalmDB (u55137) labeled with the NCBI Ebola taxonomy ID (186538)\n",
    "ebov_aa_fasta = \"/home/laura/projects/kraken2-2.1.2/ebola_rdrp_aa.fasta\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "38bc986e-19dc-42ee-b9d8-12859aeeab42",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Added \"/home/laura/projects/kraken2-2.1.2/ebola_rdrp_aa.fasta\" to library (kraken_ebov_protein)\n",
      "\tCommand being timed: \"kraken2-build --db kraken_ebov_protein --protein --no-masking --add-to-library /home/laura/projects/kraken2-2.1.2/ebola_rdrp_aa.fasta\"\n",
      "\tUser time (seconds): 0.14\n",
      "\tSystem time (seconds): 0.04\n",
      "\tPercent of CPU this job got: 45%\n",
      "\tElapsed (wall clock) time (h:mm:ss or m:ss): 0:00.41\n",
      "\tAverage shared text size (kbytes): 0\n",
      "\tAverage unshared data size (kbytes): 0\n",
      "\tAverage stack size (kbytes): 0\n",
      "\tAverage total size (kbytes): 0\n",
      "\tMaximum resident set size (kbytes): 5480\n",
      "\tAverage resident set size (kbytes): 0\n",
      "\tMajor (requiring I/O) page faults: 2\n",
      "\tMinor (reclaiming a frame) page faults: 7120\n",
      "\tVoluntary context switches: 19\n",
      "\tInvoluntary context switches: 13\n",
      "\tSwaps: 0\n",
      "\tFile system inputs: 360\n",
      "\tFile system outputs: 16\n",
      "\tSocket messages sent: 0\n",
      "\tSocket messages received: 0\n",
      "\tSignals delivered: 0\n",
      "\tPage size (bytes): 4096\n",
      "\tExit status: 0\n"
     ]
    }
   ],
   "source": [
    "# Add amino acid ebola sequence\n",
    "!cd /home/laura/projects/kraken2-2.1.2 \\\n",
    "    && /usr/bin/time -v \\\n",
    "        kraken2-build \\\n",
    "            --db kraken_ebov_protein \\\n",
    "            --protein \\\n",
    "            --no-masking \\\n",
    "            --add-to-library $ebov_aa_fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7b7eb540-2ebd-4522-84b3-65719c81fec7",
   "metadata": {},
   "source": [
    "Note: Also tried 31mers (instead of default 15mers) which lead to no alignments."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "3632c070-3222-4e04-a715-dfdddcb4f345",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Creating sequence ID to taxonomy ID map (step 1)...\n",
      "Sequence ID to taxonomy ID map complete. [0.023s]\n",
      "Estimating required capacity (step 2)...\n",
      "Estimated hash table requirement: 48272 bytes\n",
      "Capacity estimation complete. [0.045s]\n",
      "Building database files (step 3)...\n",
      "Taxonomy parsed and converted.\n",
      "CHT created with 4 bits reserved for taxid.\n",
      "Completed processing of 1 sequences, 109 aa\n",
      "Writing data to disk...  complete.\n",
      "Database files completed. [6.964s]\n",
      "Database construction complete. [Total: 7.080s]\n",
      "\tCommand being timed: \"kraken2-build --protein --db kraken_ebov_protein --build --threads 8\"\n",
      "\tUser time (seconds): 6.43\n",
      "\tSystem time (seconds): 0.98\n",
      "\tPercent of CPU this job got: 103%\n",
      "\tElapsed (wall clock) time (h:mm:ss or m:ss): 0:07.15\n",
      "\tAverage shared text size (kbytes): 0\n",
      "\tAverage unshared data size (kbytes): 0\n",
      "\tAverage stack size (kbytes): 0\n",
      "\tAverage total size (kbytes): 0\n",
      "\tMaximum resident set size (kbytes): 892948\n",
      "\tAverage resident set size (kbytes): 0\n",
      "\tMajor (requiring I/O) page faults: 9\n",
      "\tMinor (reclaiming a frame) page faults: 479308\n",
      "\tVoluntary context switches: 200\n",
      "\tInvoluntary context switches: 85\n",
      "\tSwaps: 0\n",
      "\tFile system inputs: 1056\n",
      "\tFile system outputs: 136\n",
      "\tSocket messages sent: 0\n",
      "\tSocket messages received: 0\n",
      "\tSignals delivered: 0\n",
      "\tPage size (bytes): 4096\n",
      "\tExit status: 0\n"
     ]
    }
   ],
   "source": [
    "# Build index\n",
    "!cd /home/laura/projects/kraken2-2.1.2 \\\n",
    "    && /usr/bin/time -v \\\n",
    "        kraken2-build \\\n",
    "            --protein \\\n",
    "            --db kraken_ebov_protein \\\n",
    "            --build --threads 8"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "da52c422-fab7-4b7a-b925-a4f547e1f618",
   "metadata": {},
   "source": [
    "Align:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "3016d0df-0904-431f-a2ba-0ad53ec34f8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "protein_krakendb = \"/home/laura/projects/kraken2-2.1.2/kraken_ebov_protein\"\n",
    "protein_kraken_out = f\"{out}/../kraken_mutations/mutations_protein\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "d80ee457-6daf-4b0b-837c-71ab19467db9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.020s (2062.0 Kseq/m, 181.46 Mbp/m).\n",
      "  636 sequences classified (94.08%)\n",
      "  40 sequences unclassified (5.92%)\n"
     ]
    }
   ],
   "source": [
    "# Align unmutated sequences\n",
    "orig_file = \"/home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_2_extracted_u10.fa\"\n",
    "out_file1 = \"0_ebov_snp_0_ms.k2report\"\n",
    "out_file2 = \"0_ebov_snp_0_ms.kraken2\"\n",
    "\n",
    "!~/bin/kraken2 \\\n",
    "        --db $protein_krakendb \\\n",
    "        --threads 32 \\\n",
    "        --report-minimizer-data \\\n",
    "        --report $protein_kraken_out/$out_file1 \\\n",
    "        $orig_file > $protein_kraken_out/$out_file2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "2e261027-6fc2-4b2d-9341-60bc704b9de0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2623.0 Kseq/m, 230.83 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2318.4 Kseq/m, 204.02 Mbp/m).\n",
      "  79 sequences classified (11.69%)\n",
      "  597 sequences unclassified (88.31%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2435.6 Kseq/m, 214.33 Mbp/m).\n",
      "  10 sequences classified (1.48%)\n",
      "  666 sequences unclassified (98.52%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2667.5 Kseq/m, 234.74 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2637.4 Kseq/m, 232.09 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2514.3 Kseq/m, 221.25 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3330.3 Kseq/m, 293.07 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2488.5 Kseq/m, 218.99 Mbp/m).\n",
      "  197 sequences classified (29.14%)\n",
      "  479 sequences unclassified (70.86%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2528.8 Kseq/m, 222.54 Mbp/m).\n",
      "  19 sequences classified (2.81%)\n",
      "  657 sequences unclassified (97.19%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2484.5 Kseq/m, 218.64 Mbp/m).\n",
      "  52 sequences classified (7.69%)\n",
      "  624 sequences unclassified (92.31%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2555.6 Kseq/m, 224.89 Mbp/m).\n",
      "  546 sequences classified (80.77%)\n",
      "  130 sequences unclassified (19.23%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2568.2 Kseq/m, 226.00 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2502.3 Kseq/m, 220.20 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3203.0 Kseq/m, 281.87 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3160.1 Kseq/m, 278.09 Mbp/m).\n",
      "  25 sequences classified (3.70%)\n",
      "  651 sequences unclassified (96.30%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2637.0 Kseq/m, 232.06 Mbp/m).\n",
      "  38 sequences classified (5.62%)\n",
      "  638 sequences unclassified (94.38%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2572.0 Kseq/m, 226.33 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2652.2 Kseq/m, 233.39 Mbp/m).\n",
      "  6 sequences classified (0.89%)\n",
      "  670 sequences unclassified (99.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2530.3 Kseq/m, 222.66 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2694.1 Kseq/m, 237.08 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2630.5 Kseq/m, 231.49 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2571.0 Kseq/m, 226.25 Mbp/m).\n",
      "  162 sequences classified (23.96%)\n",
      "  514 sequences unclassified (76.04%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2582.5 Kseq/m, 227.26 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2605.3 Kseq/m, 229.27 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2465.7 Kseq/m, 216.98 Mbp/m).\n",
      "  311 sequences classified (46.01%)\n",
      "  365 sequences unclassified (53.99%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2482.6 Kseq/m, 218.46 Mbp/m).\n",
      "  516 sequences classified (76.33%)\n",
      "  160 sequences unclassified (23.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2572.6 Kseq/m, 226.39 Mbp/m).\n",
      "  199 sequences classified (29.44%)\n",
      "  477 sequences unclassified (70.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2595.3 Kseq/m, 228.39 Mbp/m).\n",
      "  72 sequences classified (10.65%)\n",
      "  604 sequences unclassified (89.35%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2500.2 Kseq/m, 220.01 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3194.0 Kseq/m, 281.07 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2548.7 Kseq/m, 224.29 Mbp/m).\n",
      "  6 sequences classified (0.89%)\n",
      "  670 sequences unclassified (99.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2679.5 Kseq/m, 235.80 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2591.2 Kseq/m, 228.03 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2588.4 Kseq/m, 227.78 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2345.9 Kseq/m, 206.44 Mbp/m).\n",
      "  57 sequences classified (8.43%)\n",
      "  619 sequences unclassified (91.57%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2598.7 Kseq/m, 228.68 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3352.1 Kseq/m, 294.98 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2520.5 Kseq/m, 221.80 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.021s (1928.2 Kseq/m, 169.68 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2603.7 Kseq/m, 229.12 Mbp/m).\n",
      "  14 sequences classified (2.07%)\n",
      "  662 sequences unclassified (97.93%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3368.5 Kseq/m, 296.43 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2545.8 Kseq/m, 224.03 Mbp/m).\n",
      "  14 sequences classified (2.07%)\n",
      "  662 sequences unclassified (97.93%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2634.8 Kseq/m, 231.86 Mbp/m).\n",
      "  22 sequences classified (3.25%)\n",
      "  654 sequences unclassified (96.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2574.4 Kseq/m, 226.55 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2476.2 Kseq/m, 217.90 Mbp/m).\n",
      "  23 sequences classified (3.40%)\n",
      "  653 sequences unclassified (96.60%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2463.4 Kseq/m, 216.78 Mbp/m).\n",
      "  10 sequences classified (1.48%)\n",
      "  666 sequences unclassified (98.52%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2572.1 Kseq/m, 226.35 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2707.6 Kseq/m, 238.27 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2690.5 Kseq/m, 236.77 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2588.4 Kseq/m, 227.78 Mbp/m).\n",
      "  79 sequences classified (11.69%)\n",
      "  597 sequences unclassified (88.31%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2595.3 Kseq/m, 228.39 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2662.1 Kseq/m, 234.27 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2529.8 Kseq/m, 222.62 Mbp/m).\n",
      "  277 sequences classified (40.98%)\n",
      "  399 sequences unclassified (59.02%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3011.8 Kseq/m, 265.04 Mbp/m).\n",
      "  17 sequences classified (2.51%)\n",
      "  659 sequences unclassified (97.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2614.7 Kseq/m, 230.10 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2658.1 Kseq/m, 233.91 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2645.1 Kseq/m, 232.77 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2289.2 Kseq/m, 201.45 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2546.3 Kseq/m, 224.07 Mbp/m).\n",
      "  151 sequences classified (22.34%)\n",
      "  525 sequences unclassified (77.66%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2486.8 Kseq/m, 218.84 Mbp/m).\n",
      "  535 sequences classified (79.14%)\n",
      "  141 sequences unclassified (20.86%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2763.3 Kseq/m, 243.17 Mbp/m).\n",
      "  60 sequences classified (8.88%)\n",
      "  616 sequences unclassified (91.12%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2506.0 Kseq/m, 220.53 Mbp/m).\n",
      "  515 sequences classified (76.18%)\n",
      "  161 sequences unclassified (23.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2614.2 Kseq/m, 230.05 Mbp/m).\n",
      "  8 sequences classified (1.18%)\n",
      "  668 sequences unclassified (98.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2590.4 Kseq/m, 227.95 Mbp/m).\n",
      "  145 sequences classified (21.45%)\n",
      "  531 sequences unclassified (78.55%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2598.3 Kseq/m, 228.65 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2340.9 Kseq/m, 206.00 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2518.5 Kseq/m, 221.63 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2572.8 Kseq/m, 226.41 Mbp/m).\n",
      "  27 sequences classified (3.99%)\n",
      "  649 sequences unclassified (96.01%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2654.5 Kseq/m, 233.59 Mbp/m).\n",
      "  70 sequences classified (10.36%)\n",
      "  606 sequences unclassified (89.64%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2659.7 Kseq/m, 234.05 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2538.0 Kseq/m, 223.35 Mbp/m).\n",
      "  218 sequences classified (32.25%)\n",
      "  458 sequences unclassified (67.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2651.2 Kseq/m, 233.30 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2616.6 Kseq/m, 230.26 Mbp/m).\n",
      "  61 sequences classified (9.02%)\n",
      "  615 sequences unclassified (90.98%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2636.3 Kseq/m, 232.00 Mbp/m).\n",
      "  8 sequences classified (1.18%)\n",
      "  668 sequences unclassified (98.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2617.8 Kseq/m, 230.37 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2541.4 Kseq/m, 223.64 Mbp/m).\n",
      "  392 sequences classified (57.99%)\n",
      "  284 sequences unclassified (42.01%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2586.2 Kseq/m, 227.59 Mbp/m).\n",
      "  9 sequences classified (1.33%)\n",
      "  667 sequences unclassified (98.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2618.6 Kseq/m, 230.44 Mbp/m).\n",
      "  213 sequences classified (31.51%)\n",
      "  463 sequences unclassified (68.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2432.4 Kseq/m, 214.05 Mbp/m).\n",
      "  53 sequences classified (7.84%)\n",
      "  623 sequences unclassified (92.16%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2554.8 Kseq/m, 224.82 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2632.4 Kseq/m, 231.65 Mbp/m).\n",
      "  6 sequences classified (0.89%)\n",
      "  670 sequences unclassified (99.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2672.3 Kseq/m, 235.16 Mbp/m).\n",
      "  36 sequences classified (5.33%)\n",
      "  640 sequences unclassified (94.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2600.0 Kseq/m, 228.80 Mbp/m).\n",
      "  9 sequences classified (1.33%)\n",
      "  667 sequences unclassified (98.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2541.4 Kseq/m, 223.64 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2741.5 Kseq/m, 241.25 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2829.6 Kseq/m, 249.01 Mbp/m).\n",
      "  9 sequences classified (1.33%)\n",
      "  667 sequences unclassified (98.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.009s (4542.0 Kseq/m, 399.70 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2470.5 Kseq/m, 217.40 Mbp/m).\n",
      "  307 sequences classified (45.41%)\n",
      "  369 sequences unclassified (54.59%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2618.1 Kseq/m, 230.40 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2291.9 Kseq/m, 201.69 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2575.4 Kseq/m, 226.64 Mbp/m).\n",
      "  111 sequences classified (16.42%)\n",
      "  565 sequences unclassified (83.58%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2573.0 Kseq/m, 226.42 Mbp/m).\n",
      "  6 sequences classified (0.89%)\n",
      "  670 sequences unclassified (99.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2563.7 Kseq/m, 225.60 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2546.6 Kseq/m, 224.10 Mbp/m).\n",
      "  75 sequences classified (11.09%)\n",
      "  601 sequences unclassified (88.91%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2552.2 Kseq/m, 224.60 Mbp/m).\n",
      "  16 sequences classified (2.37%)\n",
      "  660 sequences unclassified (97.63%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2517.4 Kseq/m, 221.53 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2588.2 Kseq/m, 227.76 Mbp/m).\n",
      "  10 sequences classified (1.48%)\n",
      "  666 sequences unclassified (98.52%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2985.4 Kseq/m, 262.72 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2482.7 Kseq/m, 218.48 Mbp/m).\n",
      "  405 sequences classified (59.91%)\n",
      "  271 sequences unclassified (40.09%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2600.7 Kseq/m, 228.86 Mbp/m).\n",
      "  21 sequences classified (3.11%)\n",
      "  655 sequences unclassified (96.89%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2644.1 Kseq/m, 232.68 Mbp/m).\n",
      "  34 sequences classified (5.03%)\n",
      "  642 sequences unclassified (94.97%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2503.7 Kseq/m, 220.33 Mbp/m).\n",
      "  407 sequences classified (60.21%)\n",
      "  269 sequences unclassified (39.79%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2586.7 Kseq/m, 227.63 Mbp/m).\n",
      "  28 sequences classified (4.14%)\n",
      "  648 sequences unclassified (95.86%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2556.7 Kseq/m, 224.99 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2557.1 Kseq/m, 225.02 Mbp/m).\n",
      "  62 sequences classified (9.17%)\n",
      "  614 sequences unclassified (90.83%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2691.4 Kseq/m, 236.85 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2686.8 Kseq/m, 236.44 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2541.2 Kseq/m, 223.63 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2464.0 Kseq/m, 216.83 Mbp/m).\n",
      "  13 sequences classified (1.92%)\n",
      "  663 sequences unclassified (98.08%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2613.1 Kseq/m, 229.95 Mbp/m).\n",
      "  11 sequences classified (1.63%)\n",
      "  665 sequences unclassified (98.37%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2423.4 Kseq/m, 213.26 Mbp/m).\n",
      "  540 sequences classified (79.88%)\n",
      "  136 sequences unclassified (20.12%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2676.9 Kseq/m, 235.56 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2661.6 Kseq/m, 234.22 Mbp/m).\n",
      "  8 sequences classified (1.18%)\n",
      "  668 sequences unclassified (98.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2824.1 Kseq/m, 248.52 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2532.3 Kseq/m, 222.84 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2612.1 Kseq/m, 229.86 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3148.1 Kseq/m, 277.03 Mbp/m).\n",
      "  84 sequences classified (12.43%)\n",
      "  592 sequences unclassified (87.57%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2544.4 Kseq/m, 223.91 Mbp/m).\n",
      "  149 sequences classified (22.04%)\n",
      "  527 sequences unclassified (77.96%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2665.6 Kseq/m, 234.57 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2598.7 Kseq/m, 228.68 Mbp/m).\n",
      "  80 sequences classified (11.83%)\n",
      "  596 sequences unclassified (88.17%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3124.1 Kseq/m, 274.92 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2508.2 Kseq/m, 220.72 Mbp/m).\n",
      "  531 sequences classified (78.55%)\n",
      "  145 sequences unclassified (21.45%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2563.2 Kseq/m, 225.56 Mbp/m).\n",
      "  101 sequences classified (14.94%)\n",
      "  575 sequences unclassified (85.06%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3108.0 Kseq/m, 273.51 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2666.8 Kseq/m, 234.68 Mbp/m).\n",
      "  65 sequences classified (9.62%)\n",
      "  611 sequences unclassified (90.38%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2495.8 Kseq/m, 219.63 Mbp/m).\n",
      "  93 sequences classified (13.76%)\n",
      "  583 sequences unclassified (86.24%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2685.6 Kseq/m, 236.33 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2556.6 Kseq/m, 224.98 Mbp/m).\n",
      "  139 sequences classified (20.56%)\n",
      "  537 sequences unclassified (79.44%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2600.0 Kseq/m, 228.80 Mbp/m).\n",
      "  10 sequences classified (1.48%)\n",
      "  666 sequences unclassified (98.52%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2633.3 Kseq/m, 231.73 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2617.6 Kseq/m, 230.35 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2471.8 Kseq/m, 217.52 Mbp/m).\n",
      "  529 sequences classified (78.25%)\n",
      "  147 sequences unclassified (21.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2218.8 Kseq/m, 195.26 Mbp/m).\n",
      "  22 sequences classified (3.25%)\n",
      "  654 sequences unclassified (96.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2591.4 Kseq/m, 228.04 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2600.3 Kseq/m, 228.83 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2580.3 Kseq/m, 227.07 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3471.4 Kseq/m, 305.48 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2565.8 Kseq/m, 225.79 Mbp/m).\n",
      "  64 sequences classified (9.47%)\n",
      "  612 sequences unclassified (90.53%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2354.3 Kseq/m, 207.18 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2472.3 Kseq/m, 217.56 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2561.3 Kseq/m, 225.39 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3241.9 Kseq/m, 285.29 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2489.4 Kseq/m, 219.07 Mbp/m).\n",
      "  96 sequences classified (14.20%)\n",
      "  580 sequences unclassified (85.80%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2528.2 Kseq/m, 222.48 Mbp/m).\n",
      "  293 sequences classified (43.34%)\n",
      "  383 sequences unclassified (56.66%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2626.3 Kseq/m, 231.11 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2576.9 Kseq/m, 226.76 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2615.9 Kseq/m, 230.20 Mbp/m).\n",
      "  30 sequences classified (4.44%)\n",
      "  646 sequences unclassified (95.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2651.0 Kseq/m, 233.29 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2545.7 Kseq/m, 224.02 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2596.5 Kseq/m, 228.49 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2598.5 Kseq/m, 228.67 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2617.1 Kseq/m, 230.31 Mbp/m).\n",
      "  63 sequences classified (9.32%)\n",
      "  613 sequences unclassified (90.68%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2606.5 Kseq/m, 229.37 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2808.9 Kseq/m, 247.18 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2600.2 Kseq/m, 228.81 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2678.5 Kseq/m, 235.70 Mbp/m).\n",
      "  18 sequences classified (2.66%)\n",
      "  658 sequences unclassified (97.34%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2460.3 Kseq/m, 216.50 Mbp/m).\n",
      "  409 sequences classified (60.50%)\n",
      "  267 sequences unclassified (39.50%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2759.6 Kseq/m, 242.84 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2554.2 Kseq/m, 224.77 Mbp/m).\n",
      "  226 sequences classified (33.43%)\n",
      "  450 sequences unclassified (66.57%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2489.6 Kseq/m, 219.08 Mbp/m).\n",
      "  287 sequences classified (42.46%)\n",
      "  389 sequences unclassified (57.54%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2591.9 Kseq/m, 228.08 Mbp/m).\n",
      "  28 sequences classified (4.14%)\n",
      "  648 sequences unclassified (95.86%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2618.0 Kseq/m, 230.38 Mbp/m).\n",
      "  161 sequences classified (23.82%)\n",
      "  515 sequences unclassified (76.18%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2574.7 Kseq/m, 226.58 Mbp/m).\n",
      "  8 sequences classified (1.18%)\n",
      "  668 sequences unclassified (98.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2666.1 Kseq/m, 234.62 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2612.9 Kseq/m, 229.93 Mbp/m).\n",
      "  110 sequences classified (16.27%)\n",
      "  566 sequences unclassified (83.73%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2562.9 Kseq/m, 225.53 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2569.0 Kseq/m, 226.08 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2538.6 Kseq/m, 223.40 Mbp/m).\n",
      "  154 sequences classified (22.78%)\n",
      "  522 sequences unclassified (77.22%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2686.3 Kseq/m, 236.39 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2580.5 Kseq/m, 227.08 Mbp/m).\n",
      "  230 sequences classified (34.02%)\n",
      "  446 sequences unclassified (65.98%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3276.8 Kseq/m, 288.36 Mbp/m).\n",
      "  96 sequences classified (14.20%)\n",
      "  580 sequences unclassified (85.80%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2641.0 Kseq/m, 232.41 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2541.2 Kseq/m, 223.63 Mbp/m).\n",
      "  50 sequences classified (7.40%)\n",
      "  626 sequences unclassified (92.60%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2532.6 Kseq/m, 222.87 Mbp/m).\n",
      "  73 sequences classified (10.80%)\n",
      "  603 sequences unclassified (89.20%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2597.0 Kseq/m, 228.54 Mbp/m).\n",
      "  29 sequences classified (4.29%)\n",
      "  647 sequences unclassified (95.71%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2677.1 Kseq/m, 235.58 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2621.0 Kseq/m, 230.65 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2569.9 Kseq/m, 226.15 Mbp/m).\n",
      "  103 sequences classified (15.24%)\n",
      "  573 sequences unclassified (84.76%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.024s (1703.2 Kseq/m, 149.88 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.009s (4391.5 Kseq/m, 386.45 Mbp/m).\n",
      "  44 sequences classified (6.51%)\n",
      "  632 sequences unclassified (93.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2632.4 Kseq/m, 231.65 Mbp/m).\n",
      "  9 sequences classified (1.33%)\n",
      "  667 sequences unclassified (98.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2616.6 Kseq/m, 230.26 Mbp/m).\n",
      "  60 sequences classified (8.88%)\n",
      "  616 sequences unclassified (91.12%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2661.6 Kseq/m, 234.22 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2633.9 Kseq/m, 231.79 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2575.2 Kseq/m, 226.62 Mbp/m).\n",
      "  29 sequences classified (4.29%)\n",
      "  647 sequences unclassified (95.71%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2609.2 Kseq/m, 229.61 Mbp/m).\n",
      "  84 sequences classified (12.43%)\n",
      "  592 sequences unclassified (87.57%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2559.8 Kseq/m, 225.26 Mbp/m).\n",
      "  49 sequences classified (7.25%)\n",
      "  627 sequences unclassified (92.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2571.6 Kseq/m, 226.30 Mbp/m).\n",
      "  15 sequences classified (2.22%)\n",
      "  661 sequences unclassified (97.78%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2617.3 Kseq/m, 230.32 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2550.9 Kseq/m, 224.48 Mbp/m).\n",
      "  32 sequences classified (4.73%)\n",
      "  644 sequences unclassified (95.27%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3208.6 Kseq/m, 282.36 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2584.3 Kseq/m, 227.42 Mbp/m).\n",
      "  6 sequences classified (0.89%)\n",
      "  670 sequences unclassified (99.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2825.1 Kseq/m, 248.61 Mbp/m).\n",
      "  151 sequences classified (22.34%)\n",
      "  525 sequences unclassified (77.66%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2564.0 Kseq/m, 225.63 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2576.9 Kseq/m, 226.76 Mbp/m).\n",
      "  12 sequences classified (1.78%)\n",
      "  664 sequences unclassified (98.22%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2514.6 Kseq/m, 221.28 Mbp/m).\n",
      "  82 sequences classified (12.13%)\n",
      "  594 sequences unclassified (87.87%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2467.3 Kseq/m, 217.12 Mbp/m).\n",
      "  541 sequences classified (80.03%)\n",
      "  135 sequences unclassified (19.97%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2499.1 Kseq/m, 219.92 Mbp/m).\n",
      "  208 sequences classified (30.77%)\n",
      "  468 sequences unclassified (69.23%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2539.4 Kseq/m, 223.47 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2864.8 Kseq/m, 252.10 Mbp/m).\n",
      "  10 sequences classified (1.48%)\n",
      "  666 sequences unclassified (98.52%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2532.9 Kseq/m, 222.90 Mbp/m).\n",
      "  206 sequences classified (30.47%)\n",
      "  470 sequences unclassified (69.53%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3134.0 Kseq/m, 275.79 Mbp/m).\n",
      "  7 sequences classified (1.04%)\n",
      "  669 sequences unclassified (98.96%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2182.9 Kseq/m, 192.09 Mbp/m).\n",
      "  167 sequences classified (24.70%)\n",
      "  509 sequences unclassified (75.30%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2583.4 Kseq/m, 227.34 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2541.7 Kseq/m, 223.67 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2525.7 Kseq/m, 222.26 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2714.5 Kseq/m, 238.88 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2627.5 Kseq/m, 231.22 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2666.5 Kseq/m, 234.65 Mbp/m).\n",
      "  7 sequences classified (1.04%)\n",
      "  669 sequences unclassified (98.96%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2667.5 Kseq/m, 234.74 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3212.4 Kseq/m, 282.69 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3125.3 Kseq/m, 275.03 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2703.6 Kseq/m, 237.92 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2693.0 Kseq/m, 236.99 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2535.6 Kseq/m, 223.14 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2545.3 Kseq/m, 223.99 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2594.8 Kseq/m, 228.35 Mbp/m).\n",
      "  53 sequences classified (7.84%)\n",
      "  623 sequences unclassified (92.16%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3091.2 Kseq/m, 272.03 Mbp/m).\n",
      "  96 sequences classified (14.20%)\n",
      "  580 sequences unclassified (85.80%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2635.5 Kseq/m, 231.92 Mbp/m).\n",
      "  12 sequences classified (1.78%)\n",
      "  664 sequences unclassified (98.22%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2590.2 Kseq/m, 227.94 Mbp/m).\n",
      "  30 sequences classified (4.44%)\n",
      "  646 sequences unclassified (95.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2553.7 Kseq/m, 224.72 Mbp/m).\n",
      "  286 sequences classified (42.31%)\n",
      "  390 sequences unclassified (57.69%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2639.8 Kseq/m, 232.30 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2720.9 Kseq/m, 239.44 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2680.4 Kseq/m, 235.88 Mbp/m).\n",
      "  26 sequences classified (3.85%)\n",
      "  650 sequences unclassified (96.15%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2583.6 Kseq/m, 227.36 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2621.2 Kseq/m, 230.66 Mbp/m).\n",
      "  44 sequences classified (6.51%)\n",
      "  632 sequences unclassified (93.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2717.0 Kseq/m, 239.10 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2393.6 Kseq/m, 210.64 Mbp/m).\n",
      "  125 sequences classified (18.49%)\n",
      "  551 sequences unclassified (81.51%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2429.3 Kseq/m, 213.78 Mbp/m).\n",
      "  291 sequences classified (43.05%)\n",
      "  385 sequences unclassified (56.95%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2514.3 Kseq/m, 221.25 Mbp/m).\n",
      "  207 sequences classified (30.62%)\n",
      "  469 sequences unclassified (69.38%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2582.1 Kseq/m, 227.23 Mbp/m).\n",
      "  67 sequences classified (9.91%)\n",
      "  609 sequences unclassified (90.09%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2513.8 Kseq/m, 221.21 Mbp/m).\n",
      "  145 sequences classified (21.45%)\n",
      "  531 sequences unclassified (78.55%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2629.7 Kseq/m, 231.41 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2337.2 Kseq/m, 205.67 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2741.7 Kseq/m, 241.27 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2559.2 Kseq/m, 225.21 Mbp/m).\n",
      "  17 sequences classified (2.51%)\n",
      "  659 sequences unclassified (97.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3365.4 Kseq/m, 296.16 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2521.6 Kseq/m, 221.90 Mbp/m).\n",
      "  310 sequences classified (45.86%)\n",
      "  366 sequences unclassified (54.14%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2384.6 Kseq/m, 209.85 Mbp/m).\n",
      "  43 sequences classified (6.36%)\n",
      "  633 sequences unclassified (93.64%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3234.2 Kseq/m, 284.61 Mbp/m).\n",
      "  406 sequences classified (60.06%)\n",
      "  270 sequences unclassified (39.94%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3242.2 Kseq/m, 285.31 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2678.6 Kseq/m, 235.72 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2677.1 Kseq/m, 235.58 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3178.7 Kseq/m, 279.72 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2632.7 Kseq/m, 231.68 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2630.9 Kseq/m, 231.52 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3189.9 Kseq/m, 280.71 Mbp/m).\n",
      "  50 sequences classified (7.40%)\n",
      "  626 sequences unclassified (92.60%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3262.0 Kseq/m, 287.06 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2615.1 Kseq/m, 230.13 Mbp/m).\n",
      "  63 sequences classified (9.32%)\n",
      "  613 sequences unclassified (90.68%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2651.7 Kseq/m, 233.35 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2659.7 Kseq/m, 234.05 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2558.3 Kseq/m, 225.13 Mbp/m).\n",
      "  109 sequences classified (16.12%)\n",
      "  567 sequences unclassified (83.88%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2510.4 Kseq/m, 220.91 Mbp/m).\n",
      "  536 sequences classified (79.29%)\n",
      "  140 sequences unclassified (20.71%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2659.8 Kseq/m, 234.07 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2598.5 Kseq/m, 228.67 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2647.7 Kseq/m, 233.00 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2459.2 Kseq/m, 216.41 Mbp/m).\n",
      "  288 sequences classified (42.60%)\n",
      "  388 sequences unclassified (57.40%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3124.6 Kseq/m, 274.96 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2618.8 Kseq/m, 230.45 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2568.7 Kseq/m, 226.05 Mbp/m).\n",
      "  277 sequences classified (40.98%)\n",
      "  399 sequences unclassified (59.02%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2584.6 Kseq/m, 227.44 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2557.1 Kseq/m, 225.02 Mbp/m).\n",
      "  7 sequences classified (1.04%)\n",
      "  669 sequences unclassified (98.96%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2924.5 Kseq/m, 257.36 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2684.3 Kseq/m, 236.22 Mbp/m).\n",
      "  47 sequences classified (6.95%)\n",
      "  629 sequences unclassified (93.05%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2601.7 Kseq/m, 228.95 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2518.6 Kseq/m, 221.64 Mbp/m).\n",
      "  66 sequences classified (9.76%)\n",
      "  610 sequences unclassified (90.24%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2654.1 Kseq/m, 233.56 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3323.2 Kseq/m, 292.44 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2605.2 Kseq/m, 229.26 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2989.6 Kseq/m, 263.09 Mbp/m).\n",
      "  11 sequences classified (1.63%)\n",
      "  665 sequences unclassified (98.37%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2621.7 Kseq/m, 230.71 Mbp/m).\n",
      "  6 sequences classified (0.89%)\n",
      "  670 sequences unclassified (99.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2587.2 Kseq/m, 227.68 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3226.5 Kseq/m, 283.93 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2551.4 Kseq/m, 224.53 Mbp/m).\n",
      "  17 sequences classified (2.51%)\n",
      "  659 sequences unclassified (97.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2636.5 Kseq/m, 232.01 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2406.0 Kseq/m, 211.73 Mbp/m).\n",
      "  400 sequences classified (59.17%)\n",
      "  276 sequences unclassified (40.83%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2603.2 Kseq/m, 229.08 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2589.0 Kseq/m, 227.84 Mbp/m).\n",
      "  205 sequences classified (30.33%)\n",
      "  471 sequences unclassified (69.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2682.7 Kseq/m, 236.08 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2460.1 Kseq/m, 216.49 Mbp/m).\n",
      "  407 sequences classified (60.21%)\n",
      "  269 sequences unclassified (39.79%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3357.9 Kseq/m, 295.49 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2564.8 Kseq/m, 225.70 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3367.9 Kseq/m, 296.38 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2392.1 Kseq/m, 210.50 Mbp/m).\n",
      "  527 sequences classified (77.96%)\n",
      "  149 sequences unclassified (22.04%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3207.8 Kseq/m, 282.29 Mbp/m).\n",
      "  401 sequences classified (59.32%)\n",
      "  275 sequences unclassified (40.68%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2571.8 Kseq/m, 226.32 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2651.8 Kseq/m, 233.36 Mbp/m).\n",
      "  47 sequences classified (6.95%)\n",
      "  629 sequences unclassified (93.05%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2481.3 Kseq/m, 218.36 Mbp/m).\n",
      "  404 sequences classified (59.76%)\n",
      "  272 sequences unclassified (40.24%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2559.2 Kseq/m, 225.21 Mbp/m).\n",
      "  405 sequences classified (59.91%)\n",
      "  271 sequences unclassified (40.09%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2572.0 Kseq/m, 226.33 Mbp/m).\n",
      "  77 sequences classified (11.39%)\n",
      "  599 sequences unclassified (88.61%)\n"
     ]
    }
   ],
   "source": [
    "for mut_file in glob.glob(f\"{out}/*_ms.fa\"):\n",
    "    out_file1 = mut_file.split(\"/\")[-1].split(\".fa\")[0] + \".k2report\"\n",
    "    out_file2 = mut_file.split(\"/\")[-1].split(\".fa\")[0] + \".kraken2\"\n",
    "    \n",
    "    !~/bin/kraken2 \\\n",
    "            --db $protein_krakendb \\\n",
    "            --threads 32 \\\n",
    "            --report-minimizer-data \\\n",
    "            --report $protein_kraken_out/$out_file1 \\\n",
    "            $mut_file > $protein_kraken_out/$out_file2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27f820a3-4e92-405e-94b5-28a57d0d326d",
   "metadata": {},
   "source": [
    "### Align using DNA Kraken2:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "424e3247-5ab5-487a-9c92-68bb81c93865",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading nucleotide gb accession to taxon map... done.\n",
      "Downloading nucleotide wgs accession to taxon map... done.\n",
      "Downloaded accession to taxon map(s)\n",
      "Downloading taxonomy tree data... done.\n",
      "Uncompressing taxonomy data... done.\n",
      "Untarring taxonomy tree data... done.\n",
      "Step 1/2: Performing rsync file transfer of requested files\n",
      "Rsync file transfer complete.\n",
      "Step 2/2: Assigning taxonomic IDs to sequences\n",
      "Processed 12345 projects (15404 sequences, 505.71 Mbp)... done.\n",
      "All files processed, cleaning up extra sequence files... done, library complete.\n",
      "Masking low-complexity regions of downloaded library...which: no dustmasker in (/home/laura/projects/kraken2-2.1.2:/home/laura/bin:/home/laura/bin:/home/laura/anaconda3/bin:/home/laura/anaconda3/condabin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/laura/.local/bin:/home/laura/bin:/home/laura/.local/bin:/home/laura/bin)\n",
      "Unable to find dustmasker in path, can't mask low-complexity sequences\n",
      "Masking low-complexity regions of new file...which: no dustmasker in (/home/laura/projects/kraken2-2.1.2:/home/laura/bin:/home/laura/bin:/home/laura/anaconda3/bin:/home/laura/anaconda3/condabin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/laura/.local/bin:/home/laura/bin:/home/laura/.local/bin:/home/laura/bin)\n",
      "Unable to find dustmasker in path, can't mask low-complexity sequences\n",
      "Creating sequence ID to taxonomy ID map (step 1)...\n",
      "Found 1/1 targets, searched through 231029360 accession IDs, search complete.\n",
      "Sequence ID to taxonomy ID map complete. [18.852s]\n",
      "Estimating required capacity (step 2)...\n",
      "Estimated hash table requirement: 603293988 bytes\n",
      "Capacity estimation complete. [2.798s]\n",
      "Building database files (step 3)...\n",
      "Taxonomy parsed and converted.\n",
      "CHT created with 15 bits reserved for taxid.\n",
      "Completed processing of 15404 sequences, 505731868 bp\n",
      "Writing data to disk...  complete.\n",
      "Database files completed. [1m57.710s]\n",
      "Database construction complete. [Total: 2m19.397s]\n"
     ]
    }
   ],
   "source": [
    "# Build viral Kraken index with added EBOV genome\n",
    "!cd /home/laura/projects/kraken2-2.1.2 && kraken2-build --db krakendb --download-taxonomy\n",
    "!cd /home/laura/projects/kraken2-2.1.2 && kraken2-build --db krakendb --download-library viral\n",
    "!cd /home/laura/projects/kraken2-2.1.2 && kraken2-build --db krakendb --add-to-library ebola_genome.fa\n",
    "!cd /home/laura/projects/kraken2-2.1.2 && kraken2-build --db krakendb --build --threads 8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ca39fc6b-61dc-4c2b-ac97-448886b6c850",
   "metadata": {},
   "outputs": [],
   "source": [
    "krakendb = \"/home/laura/projects/kraken2-2.1.2/krakendb\"\n",
    "kraken_out = f\"{out}/../kraken_mutations/mutations_nn\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "93681b5d-b20f-4869-b912-08a2c4ff6fdc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.009s (4425.0 Kseq/m, 389.40 Mbp/m).\n",
      "  666 sequences classified (98.52%)\n",
      "  10 sequences unclassified (1.48%)\n"
     ]
    }
   ],
   "source": [
    "# Align unmutated sequences\n",
    "orig_file = \"/home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_2_extracted_u10.fa\"\n",
    "out_file1 = \"0_ebov_snp_0_ms.k2report\"\n",
    "out_file2 = \"0_ebov_snp_0_ms.kraken2\"\n",
    "\n",
    "!~/bin/kraken2 \\\n",
    "        --db $krakendb \\\n",
    "        --threads 32 \\\n",
    "        --report-minimizer-data \\\n",
    "        --report $kraken_out/$out_file1 \\\n",
    "        $orig_file > $kraken_out/$out_file2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "614ae502-dc1c-4097-8332-dd822325511f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2228.3 Kseq/m, 196.09 Mbp/m).\n",
      "  6 sequences classified (0.89%)\n",
      "  670 sequences unclassified (99.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2139.7 Kseq/m, 188.29 Mbp/m).\n",
      "  210 sequences classified (31.07%)\n",
      "  466 sequences unclassified (68.93%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.009s (4577.4 Kseq/m, 402.81 Mbp/m).\n",
      "  33 sequences classified (4.88%)\n",
      "  643 sequences unclassified (95.12%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.008s (4907.4 Kseq/m, 431.85 Mbp/m).\n",
      "  16 sequences classified (2.37%)\n",
      "  660 sequences unclassified (97.63%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2500.8 Kseq/m, 220.07 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.008s (5067.5 Kseq/m, 445.94 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2481.5 Kseq/m, 218.37 Mbp/m).\n",
      "  14 sequences classified (2.07%)\n",
      "  662 sequences unclassified (97.93%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3126.5 Kseq/m, 275.13 Mbp/m).\n",
      "  461 sequences classified (68.20%)\n",
      "  215 sequences unclassified (31.80%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2598.7 Kseq/m, 228.68 Mbp/m).\n",
      "  51 sequences classified (7.54%)\n",
      "  625 sequences unclassified (92.46%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3968.3 Kseq/m, 349.21 Mbp/m).\n",
      "  147 sequences classified (21.75%)\n",
      "  529 sequences unclassified (78.25%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3961.3 Kseq/m, 348.60 Mbp/m).\n",
      "  645 sequences classified (95.41%)\n",
      "  31 sequences unclassified (4.59%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2269.3 Kseq/m, 199.70 Mbp/m).\n",
      "  19 sequences classified (2.81%)\n",
      "  657 sequences unclassified (97.19%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2646.3 Kseq/m, 232.88 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2418.7 Kseq/m, 212.85 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.066s (615.5 Kseq/m, 54.16 Mbp/m).\n",
      "  78 sequences classified (11.54%)\n",
      "  598 sequences unclassified (88.46%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2307.7 Kseq/m, 203.08 Mbp/m).\n",
      "  116 sequences classified (17.16%)\n",
      "  560 sequences unclassified (82.84%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.020s (2048.2 Kseq/m, 180.24 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2174.6 Kseq/m, 191.36 Mbp/m).\n",
      "  19 sequences classified (2.81%)\n",
      "  657 sequences unclassified (97.19%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2996.5 Kseq/m, 263.69 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2458.0 Kseq/m, 216.31 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2632.7 Kseq/m, 231.68 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.008s (4887.3 Kseq/m, 430.09 Mbp/m).\n",
      "  375 sequences classified (55.47%)\n",
      "  301 sequences unclassified (44.53%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2478.6 Kseq/m, 218.12 Mbp/m).\n",
      "  11 sequences classified (1.63%)\n",
      "  665 sequences unclassified (98.37%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2442.6 Kseq/m, 214.95 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.042s (958.2 Kseq/m, 84.32 Mbp/m).\n",
      "  530 sequences classified (78.40%)\n",
      "  146 sequences unclassified (21.60%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2357.0 Kseq/m, 207.42 Mbp/m).\n",
      "  647 sequences classified (95.71%)\n",
      "  29 sequences unclassified (4.29%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2183.7 Kseq/m, 192.17 Mbp/m).\n",
      "  475 sequences classified (70.27%)\n",
      "  201 sequences unclassified (29.73%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2139.5 Kseq/m, 188.27 Mbp/m).\n",
      "  205 sequences classified (30.33%)\n",
      "  471 sequences unclassified (69.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2910.4 Kseq/m, 256.12 Mbp/m).\n",
      "  10 sequences classified (1.48%)\n",
      "  666 sequences unclassified (98.52%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2181.3 Kseq/m, 191.96 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2267.3 Kseq/m, 199.52 Mbp/m).\n",
      "  8 sequences classified (1.18%)\n",
      "  668 sequences unclassified (98.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3372.4 Kseq/m, 296.77 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2795.9 Kseq/m, 246.04 Mbp/m).\n",
      "  6 sequences classified (0.89%)\n",
      "  670 sequences unclassified (99.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3463.1 Kseq/m, 304.75 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3327.0 Kseq/m, 292.78 Mbp/m).\n",
      "  215 sequences classified (31.80%)\n",
      "  461 sequences unclassified (68.20%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2189.7 Kseq/m, 192.69 Mbp/m).\n",
      "  12 sequences classified (1.78%)\n",
      "  664 sequences unclassified (98.22%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2293.1 Kseq/m, 201.79 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2205.1 Kseq/m, 194.05 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.049s (833.8 Kseq/m, 73.38 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2170.5 Kseq/m, 191.00 Mbp/m).\n",
      "  36 sequences classified (5.33%)\n",
      "  640 sequences unclassified (94.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4034.2 Kseq/m, 355.01 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2105.4 Kseq/m, 185.27 Mbp/m).\n",
      "  30 sequences classified (4.44%)\n",
      "  646 sequences unclassified (95.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2203.6 Kseq/m, 193.92 Mbp/m).\n",
      "  71 sequences classified (10.50%)\n",
      "  605 sequences unclassified (89.50%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3522.1 Kseq/m, 309.94 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2320.1 Kseq/m, 204.17 Mbp/m).\n",
      "  60 sequences classified (8.88%)\n",
      "  616 sequences unclassified (91.12%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.033s (1219.5 Kseq/m, 107.32 Mbp/m).\n",
      "  38 sequences classified (5.62%)\n",
      "  638 sequences unclassified (94.38%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2963.6 Kseq/m, 260.80 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4184.0 Kseq/m, 368.19 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2343.3 Kseq/m, 206.21 Mbp/m).\n",
      "  9 sequences classified (1.33%)\n",
      "  667 sequences unclassified (98.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2462.2 Kseq/m, 216.67 Mbp/m).\n",
      "  217 sequences classified (32.10%)\n",
      "  459 sequences unclassified (67.90%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3367.9 Kseq/m, 296.38 Mbp/m).\n",
      "  6 sequences classified (0.89%)\n",
      "  670 sequences unclassified (99.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2571.2 Kseq/m, 226.26 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3626.9 Kseq/m, 319.17 Mbp/m).\n",
      "  543 sequences classified (80.33%)\n",
      "  133 sequences unclassified (19.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3419.6 Kseq/m, 300.93 Mbp/m).\n",
      "  44 sequences classified (6.51%)\n",
      "  632 sequences unclassified (93.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2121.0 Kseq/m, 186.65 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2194.9 Kseq/m, 193.15 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2132.4 Kseq/m, 187.65 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2165.2 Kseq/m, 190.53 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2358.7 Kseq/m, 207.56 Mbp/m).\n",
      "  355 sequences classified (52.51%)\n",
      "  321 sequences unclassified (47.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3881.3 Kseq/m, 341.56 Mbp/m).\n",
      "  652 sequences classified (96.45%)\n",
      "  24 sequences unclassified (3.55%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2161.0 Kseq/m, 190.17 Mbp/m).\n",
      "  151 sequences classified (22.34%)\n",
      "  525 sequences unclassified (77.66%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2369.6 Kseq/m, 208.52 Mbp/m).\n",
      "  650 sequences classified (96.15%)\n",
      "  26 sequences unclassified (3.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2181.9 Kseq/m, 192.01 Mbp/m).\n",
      "  11 sequences classified (1.63%)\n",
      "  665 sequences unclassified (98.37%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.022s (1882.3 Kseq/m, 165.64 Mbp/m).\n",
      "  354 sequences classified (52.37%)\n",
      "  322 sequences unclassified (47.63%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2562.2 Kseq/m, 225.48 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3475.3 Kseq/m, 305.82 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3494.1 Kseq/m, 307.48 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2116.8 Kseq/m, 186.28 Mbp/m).\n",
      "  75 sequences classified (11.09%)\n",
      "  601 sequences unclassified (88.91%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2280.7 Kseq/m, 200.70 Mbp/m).\n",
      "  205 sequences classified (30.33%)\n",
      "  471 sequences unclassified (69.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2139.1 Kseq/m, 188.24 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2157.0 Kseq/m, 189.81 Mbp/m).\n",
      "  454 sequences classified (67.16%)\n",
      "  222 sequences unclassified (32.84%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2213.1 Kseq/m, 194.76 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2808.1 Kseq/m, 247.11 Mbp/m).\n",
      "  157 sequences classified (23.22%)\n",
      "  519 sequences unclassified (76.78%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3383.7 Kseq/m, 297.76 Mbp/m).\n",
      "  12 sequences classified (1.78%)\n",
      "  664 sequences unclassified (98.22%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2317.2 Kseq/m, 203.91 Mbp/m).\n",
      "  17 sequences classified (2.51%)\n",
      "  659 sequences unclassified (97.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2157.0 Kseq/m, 189.81 Mbp/m).\n",
      "  611 sequences classified (90.38%)\n",
      "  65 sequences unclassified (9.62%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2135.6 Kseq/m, 187.94 Mbp/m).\n",
      "  27 sequences classified (3.99%)\n",
      "  649 sequences unclassified (96.01%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.025s (1633.0 Kseq/m, 143.70 Mbp/m).\n",
      "  451 sequences classified (66.72%)\n",
      "  225 sequences unclassified (33.28%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3796.3 Kseq/m, 334.08 Mbp/m).\n",
      "  160 sequences classified (23.67%)\n",
      "  516 sequences unclassified (76.33%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2469.7 Kseq/m, 217.33 Mbp/m).\n",
      "  10 sequences classified (1.48%)\n",
      "  666 sequences unclassified (98.52%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.021s (1959.3 Kseq/m, 172.42 Mbp/m).\n",
      "  14 sequences classified (2.07%)\n",
      "  662 sequences unclassified (97.93%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2116.9 Kseq/m, 186.29 Mbp/m).\n",
      "  109 sequences classified (16.12%)\n",
      "  567 sequences unclassified (83.88%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4028.6 Kseq/m, 354.52 Mbp/m).\n",
      "  19 sequences classified (2.81%)\n",
      "  657 sequences unclassified (97.19%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3777.6 Kseq/m, 332.43 Mbp/m).\n",
      "  12 sequences classified (1.78%)\n",
      "  664 sequences unclassified (98.22%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3510.5 Kseq/m, 308.92 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2167.6 Kseq/m, 190.75 Mbp/m).\n",
      "  50 sequences classified (7.40%)\n",
      "  626 sequences unclassified (92.60%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2108.5 Kseq/m, 185.55 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3524.8 Kseq/m, 310.18 Mbp/m).\n",
      "  542 sequences classified (80.18%)\n",
      "  134 sequences unclassified (19.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2610.4 Kseq/m, 229.71 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.009s (4690.6 Kseq/m, 412.78 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3489.0 Kseq/m, 307.03 Mbp/m).\n",
      "  286 sequences classified (42.31%)\n",
      "  390 sequences unclassified (57.69%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2147.6 Kseq/m, 188.99 Mbp/m).\n",
      "  14 sequences classified (2.07%)\n",
      "  662 sequences unclassified (97.93%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3719.4 Kseq/m, 327.31 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2230.5 Kseq/m, 196.29 Mbp/m).\n",
      "  210 sequences classified (31.07%)\n",
      "  466 sequences unclassified (68.93%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3025.3 Kseq/m, 266.23 Mbp/m).\n",
      "  33 sequences classified (4.88%)\n",
      "  643 sequences unclassified (95.12%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2325.2 Kseq/m, 204.61 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2235.2 Kseq/m, 196.70 Mbp/m).\n",
      "  53 sequences classified (7.84%)\n",
      "  623 sequences unclassified (92.16%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2220.9 Kseq/m, 195.44 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.055s (738.0 Kseq/m, 64.94 Mbp/m).\n",
      "  612 sequences classified (90.53%)\n",
      "  64 sequences unclassified (9.47%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3255.0 Kseq/m, 286.44 Mbp/m).\n",
      "  73 sequences classified (10.80%)\n",
      "  603 sequences unclassified (89.20%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2161.5 Kseq/m, 190.21 Mbp/m).\n",
      "  124 sequences classified (18.34%)\n",
      "  552 sequences unclassified (81.66%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2242.5 Kseq/m, 197.34 Mbp/m).\n",
      "  615 sequences classified (90.98%)\n",
      "  61 sequences unclassified (9.02%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2383.2 Kseq/m, 209.72 Mbp/m).\n",
      "  106 sequences classified (15.68%)\n",
      "  570 sequences unclassified (84.32%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2297.0 Kseq/m, 202.13 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2333.9 Kseq/m, 205.38 Mbp/m).\n",
      "  202 sequences classified (29.88%)\n",
      "  474 sequences unclassified (70.12%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2975.6 Kseq/m, 261.85 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3415.0 Kseq/m, 300.52 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2165.9 Kseq/m, 190.60 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2186.2 Kseq/m, 192.38 Mbp/m).\n",
      "  40 sequences classified (5.92%)\n",
      "  636 sequences unclassified (94.08%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2154.1 Kseq/m, 189.56 Mbp/m).\n",
      "  29 sequences classified (4.29%)\n",
      "  647 sequences unclassified (95.71%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2744.4 Kseq/m, 241.51 Mbp/m).\n",
      "  650 sequences classified (96.15%)\n",
      "  26 sequences unclassified (3.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2784.2 Kseq/m, 245.01 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.043s (950.2 Kseq/m, 83.62 Mbp/m).\n",
      "  27 sequences classified (3.99%)\n",
      "  649 sequences unclassified (96.01%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2376.2 Kseq/m, 209.11 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2248.5 Kseq/m, 197.86 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3586.5 Kseq/m, 315.61 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3887.3 Kseq/m, 342.08 Mbp/m).\n",
      "  220 sequences classified (32.54%)\n",
      "  456 sequences unclassified (67.46%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2270.4 Kseq/m, 199.79 Mbp/m).\n",
      "  358 sequences classified (52.96%)\n",
      "  318 sequences unclassified (47.04%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3908.3 Kseq/m, 343.93 Mbp/m).\n",
      "  9 sequences classified (1.33%)\n",
      "  667 sequences unclassified (98.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2321.0 Kseq/m, 204.25 Mbp/m).\n",
      "  211 sequences classified (31.21%)\n",
      "  465 sequences unclassified (68.79%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4021.4 Kseq/m, 353.88 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2206.6 Kseq/m, 194.18 Mbp/m).\n",
      "  649 sequences classified (96.01%)\n",
      "  27 sequences unclassified (3.99%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3942.8 Kseq/m, 346.97 Mbp/m).\n",
      "  280 sequences classified (41.42%)\n",
      "  396 sequences unclassified (58.58%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2207.3 Kseq/m, 194.25 Mbp/m).\n",
      "  17 sequences classified (2.51%)\n",
      "  659 sequences unclassified (97.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3816.7 Kseq/m, 335.87 Mbp/m).\n",
      "  219 sequences classified (32.40%)\n",
      "  457 sequences unclassified (67.60%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3856.2 Kseq/m, 339.35 Mbp/m).\n",
      "  264 sequences classified (39.05%)\n",
      "  412 sequences unclassified (60.95%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3942.5 Kseq/m, 346.94 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3641.3 Kseq/m, 320.43 Mbp/m).\n",
      "  373 sequences classified (55.18%)\n",
      "  303 sequences unclassified (44.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3023.7 Kseq/m, 266.09 Mbp/m).\n",
      "  51 sequences classified (7.54%)\n",
      "  625 sequences unclassified (92.46%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2607.4 Kseq/m, 229.45 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3774.1 Kseq/m, 332.12 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2475.0 Kseq/m, 217.80 Mbp/m).\n",
      "  642 sequences classified (94.97%)\n",
      "  34 sequences unclassified (5.03%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3631.2 Kseq/m, 319.54 Mbp/m).\n",
      "  44 sequences classified (6.51%)\n",
      "  632 sequences unclassified (93.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2968.8 Kseq/m, 261.26 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2396.6 Kseq/m, 210.90 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3827.9 Kseq/m, 336.85 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2511.8 Kseq/m, 221.04 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3537.7 Kseq/m, 311.32 Mbp/m).\n",
      "  206 sequences classified (30.47%)\n",
      "  470 sequences unclassified (69.53%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2576.1 Kseq/m, 226.69 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2415.7 Kseq/m, 212.58 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.051s (800.2 Kseq/m, 70.42 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2515.0 Kseq/m, 221.32 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2355.8 Kseq/m, 207.31 Mbp/m).\n",
      "  279 sequences classified (41.27%)\n",
      "  397 sequences unclassified (58.73%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3512.0 Kseq/m, 309.06 Mbp/m).\n",
      "  536 sequences classified (79.29%)\n",
      "  140 sequences unclassified (20.71%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2192.0 Kseq/m, 192.89 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2149.9 Kseq/m, 189.19 Mbp/m).\n",
      "  14 sequences classified (2.07%)\n",
      "  662 sequences unclassified (97.93%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3855.5 Kseq/m, 339.29 Mbp/m).\n",
      "  84 sequences classified (12.43%)\n",
      "  592 sequences unclassified (87.57%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2277.4 Kseq/m, 200.41 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2910.0 Kseq/m, 256.08 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2503.7 Kseq/m, 220.33 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2213.2 Kseq/m, 194.77 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3439.0 Kseq/m, 302.64 Mbp/m).\n",
      "  199 sequences classified (29.44%)\n",
      "  477 sequences unclassified (70.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2329.6 Kseq/m, 205.00 Mbp/m).\n",
      "  30 sequences classified (4.44%)\n",
      "  646 sequences unclassified (95.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2167.0 Kseq/m, 190.70 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.039s (1031.6 Kseq/m, 90.78 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.009s (4570.7 Kseq/m, 402.22 Mbp/m).\n",
      "  74 sequences classified (10.95%)\n",
      "  602 sequences unclassified (89.05%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2442.2 Kseq/m, 214.91 Mbp/m).\n",
      "  612 sequences classified (90.53%)\n",
      "  64 sequences unclassified (9.47%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3011.1 Kseq/m, 264.98 Mbp/m).\n",
      "  15 sequences classified (2.22%)\n",
      "  661 sequences unclassified (97.78%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.009s (4673.9 Kseq/m, 411.30 Mbp/m).\n",
      "  466 sequences classified (68.93%)\n",
      "  210 sequences unclassified (31.07%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.008s (4898.6 Kseq/m, 431.07 Mbp/m).\n",
      "  535 sequences classified (79.14%)\n",
      "  141 sequences unclassified (20.86%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2977.1 Kseq/m, 261.98 Mbp/m).\n",
      "  77 sequences classified (11.39%)\n",
      "  599 sequences unclassified (88.61%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2148.2 Kseq/m, 189.04 Mbp/m).\n",
      "  365 sequences classified (53.99%)\n",
      "  311 sequences unclassified (46.01%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2155.3 Kseq/m, 189.66 Mbp/m).\n",
      "  13 sequences classified (1.92%)\n",
      "  663 sequences unclassified (98.08%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2402.8 Kseq/m, 211.45 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3018.1 Kseq/m, 265.59 Mbp/m).\n",
      "  278 sequences classified (41.12%)\n",
      "  398 sequences unclassified (58.88%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2197.3 Kseq/m, 193.36 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2182.1 Kseq/m, 192.02 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2204.5 Kseq/m, 193.99 Mbp/m).\n",
      "  399 sequences classified (59.02%)\n",
      "  277 sequences unclassified (40.98%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3496.6 Kseq/m, 307.70 Mbp/m).\n",
      "  27 sequences classified (3.99%)\n",
      "  649 sequences unclassified (96.01%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.032s (1282.0 Kseq/m, 112.81 Mbp/m).\n",
      "  471 sequences classified (69.67%)\n",
      "  205 sequences unclassified (30.33%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3970.6 Kseq/m, 349.42 Mbp/m).\n",
      "  280 sequences classified (41.42%)\n",
      "  396 sequences unclassified (58.58%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2833.8 Kseq/m, 249.37 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2159.7 Kseq/m, 190.06 Mbp/m).\n",
      "  107 sequences classified (15.83%)\n",
      "  569 sequences unclassified (84.17%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3428.6 Kseq/m, 301.71 Mbp/m).\n",
      "  191 sequences classified (28.25%)\n",
      "  485 sequences unclassified (71.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3301.3 Kseq/m, 290.52 Mbp/m).\n",
      "  70 sequences classified (10.36%)\n",
      "  606 sequences unclassified (89.64%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2174.7 Kseq/m, 191.37 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.057s (710.3 Kseq/m, 62.50 Mbp/m).\n",
      "  16 sequences classified (2.37%)\n",
      "  660 sequences unclassified (97.63%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3528.2 Kseq/m, 310.48 Mbp/m).\n",
      "  279 sequences classified (41.27%)\n",
      "  397 sequences unclassified (58.73%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2154.0 Kseq/m, 189.55 Mbp/m).\n",
      "  17 sequences classified (2.51%)\n",
      "  659 sequences unclassified (97.49%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.009s (4407.3 Kseq/m, 387.84 Mbp/m).\n",
      "  158 sequences classified (23.37%)\n",
      "  518 sequences unclassified (76.63%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.009s (4745.0 Kseq/m, 417.56 Mbp/m).\n",
      "  35 sequences classified (5.18%)\n",
      "  641 sequences unclassified (94.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2192.8 Kseq/m, 192.97 Mbp/m).\n",
      "  138 sequences classified (20.41%)\n",
      "  538 sequences unclassified (79.59%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2981.5 Kseq/m, 262.37 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.057s (712.3 Kseq/m, 62.68 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3810.6 Kseq/m, 335.33 Mbp/m).\n",
      "  76 sequences classified (11.24%)\n",
      "  600 sequences unclassified (88.76%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4087.9 Kseq/m, 359.73 Mbp/m).\n",
      "  204 sequences classified (30.18%)\n",
      "  472 sequences unclassified (69.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2506.5 Kseq/m, 220.57 Mbp/m).\n",
      "  148 sequences classified (21.89%)\n",
      "  528 sequences unclassified (78.11%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2342.5 Kseq/m, 206.14 Mbp/m).\n",
      "  46 sequences classified (6.80%)\n",
      "  630 sequences unclassified (93.20%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2349.0 Kseq/m, 206.71 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3949.0 Kseq/m, 347.51 Mbp/m).\n",
      "  110 sequences classified (16.27%)\n",
      "  566 sequences unclassified (83.73%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.043s (953.2 Kseq/m, 83.88 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3538.0 Kseq/m, 311.35 Mbp/m).\n",
      "  13 sequences classified (1.92%)\n",
      "  663 sequences unclassified (98.08%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3543.6 Kseq/m, 311.84 Mbp/m).\n",
      "  369 sequences classified (54.59%)\n",
      "  307 sequences unclassified (45.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4043.5 Kseq/m, 355.82 Mbp/m).\n",
      "  28 sequences classified (4.14%)\n",
      "  648 sequences unclassified (95.86%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2283.3 Kseq/m, 200.93 Mbp/m).\n",
      "  40 sequences classified (5.92%)\n",
      "  636 sequences unclassified (94.08%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2416.0 Kseq/m, 212.61 Mbp/m).\n",
      "  206 sequences classified (30.47%)\n",
      "  470 sequences unclassified (69.53%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2428.3 Kseq/m, 213.69 Mbp/m).\n",
      "  650 sequences classified (96.15%)\n",
      "  26 sequences unclassified (3.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2474.1 Kseq/m, 217.72 Mbp/m).\n",
      "  467 sequences classified (69.08%)\n",
      "  209 sequences unclassified (30.92%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2273.9 Kseq/m, 200.11 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2399.4 Kseq/m, 211.15 Mbp/m).\n",
      "  28 sequences classified (4.14%)\n",
      "  648 sequences unclassified (95.86%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.022s (1826.8 Kseq/m, 160.76 Mbp/m).\n",
      "  436 sequences classified (64.50%)\n",
      "  240 sequences unclassified (35.50%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3269.1 Kseq/m, 287.68 Mbp/m).\n",
      "  15 sequences classified (2.22%)\n",
      "  661 sequences unclassified (97.78%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3148.3 Kseq/m, 277.05 Mbp/m).\n",
      "  372 sequences classified (55.03%)\n",
      "  304 sequences unclassified (44.97%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2362.8 Kseq/m, 207.93 Mbp/m).\n",
      "  14 sequences classified (2.07%)\n",
      "  662 sequences unclassified (97.93%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.033s (1223.6 Kseq/m, 107.67 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2202.7 Kseq/m, 193.84 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2401.0 Kseq/m, 211.29 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2289.6 Kseq/m, 201.48 Mbp/m).\n",
      "  12 sequences classified (1.78%)\n",
      "  664 sequences unclassified (98.22%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.020s (2034.4 Kseq/m, 179.03 Mbp/m).\n",
      "  22 sequences classified (3.25%)\n",
      "  654 sequences unclassified (96.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2142.6 Kseq/m, 188.55 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4109.4 Kseq/m, 361.63 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2516.6 Kseq/m, 221.46 Mbp/m).\n",
      "  12 sequences classified (1.78%)\n",
      "  664 sequences unclassified (98.22%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3923.0 Kseq/m, 345.22 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3926.8 Kseq/m, 345.56 Mbp/m).\n",
      "  7 sequences classified (1.04%)\n",
      "  669 sequences unclassified (98.96%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2319.4 Kseq/m, 204.11 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2279.5 Kseq/m, 200.60 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4104.0 Kseq/m, 361.15 Mbp/m).\n",
      "  139 sequences classified (20.56%)\n",
      "  537 sequences unclassified (79.44%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2338.7 Kseq/m, 205.81 Mbp/m).\n",
      "  279 sequences classified (41.27%)\n",
      "  397 sequences unclassified (58.73%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4022.6 Kseq/m, 353.99 Mbp/m).\n",
      "  45 sequences classified (6.66%)\n",
      "  631 sequences unclassified (93.34%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3678.9 Kseq/m, 323.74 Mbp/m).\n",
      "  90 sequences classified (13.31%)\n",
      "  586 sequences unclassified (86.69%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4037.8 Kseq/m, 355.33 Mbp/m).\n",
      "  544 sequences classified (80.47%)\n",
      "  132 sequences unclassified (19.53%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2368.6 Kseq/m, 208.44 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2565.8 Kseq/m, 225.79 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2302.7 Kseq/m, 202.64 Mbp/m).\n",
      "  74 sequences classified (10.95%)\n",
      "  602 sequences unclassified (89.05%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3776.5 Kseq/m, 332.34 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2387.6 Kseq/m, 210.11 Mbp/m).\n",
      "  145 sequences classified (21.45%)\n",
      "  531 sequences unclassified (78.55%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3974.9 Kseq/m, 349.79 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2507.4 Kseq/m, 220.65 Mbp/m).\n",
      "  275 sequences classified (40.68%)\n",
      "  401 sequences unclassified (59.32%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.065s (622.0 Kseq/m, 54.73 Mbp/m).\n",
      "  533 sequences classified (78.85%)\n",
      "  143 sequences unclassified (21.15%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3656.0 Kseq/m, 321.73 Mbp/m).\n",
      "  428 sequences classified (63.31%)\n",
      "  248 sequences unclassified (36.69%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3666.3 Kseq/m, 322.63 Mbp/m).\n",
      "  191 sequences classified (28.25%)\n",
      "  485 sequences unclassified (71.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2342.2 Kseq/m, 206.11 Mbp/m).\n",
      "  339 sequences classified (50.15%)\n",
      "  337 sequences unclassified (49.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4036.6 Kseq/m, 355.22 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2194.1 Kseq/m, 193.08 Mbp/m).\n",
      "  16 sequences classified (2.37%)\n",
      "  660 sequences unclassified (97.63%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2585.9 Kseq/m, 227.56 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.080s (508.2 Kseq/m, 44.72 Mbp/m).\n",
      "  56 sequences classified (8.28%)\n",
      "  620 sequences unclassified (91.72%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3963.3 Kseq/m, 348.77 Mbp/m).\n",
      "  9 sequences classified (1.33%)\n",
      "  667 sequences unclassified (98.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3651.1 Kseq/m, 321.30 Mbp/m).\n",
      "  536 sequences classified (79.29%)\n",
      "  140 sequences unclassified (20.71%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3996.5 Kseq/m, 351.69 Mbp/m).\n",
      "  100 sequences classified (14.79%)\n",
      "  576 sequences unclassified (85.21%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2476.5 Kseq/m, 217.93 Mbp/m).\n",
      "  611 sequences classified (90.38%)\n",
      "  65 sequences unclassified (9.62%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2422.1 Kseq/m, 213.14 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2613.1 Kseq/m, 229.95 Mbp/m).\n",
      "  22 sequences classified (3.25%)\n",
      "  654 sequences unclassified (96.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2198.0 Kseq/m, 193.43 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2389.0 Kseq/m, 210.23 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2617.4 Kseq/m, 230.34 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2325.8 Kseq/m, 204.67 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2176.4 Kseq/m, 191.53 Mbp/m).\n",
      "  198 sequences classified (29.29%)\n",
      "  478 sequences unclassified (70.71%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2671.2 Kseq/m, 235.07 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3951.7 Kseq/m, 347.75 Mbp/m).\n",
      "  187 sequences classified (27.66%)\n",
      "  489 sequences unclassified (72.34%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.033s (1219.4 Kseq/m, 107.30 Mbp/m).\n",
      "  31 sequences classified (4.59%)\n",
      "  645 sequences unclassified (95.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2309.9 Kseq/m, 203.27 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3127.7 Kseq/m, 275.24 Mbp/m).\n",
      "  280 sequences classified (41.42%)\n",
      "  396 sequences unclassified (58.58%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3563.5 Kseq/m, 313.59 Mbp/m).\n",
      "  642 sequences classified (94.97%)\n",
      "  34 sequences unclassified (5.03%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2560.1 Kseq/m, 225.29 Mbp/m).\n",
      "  20 sequences classified (2.96%)\n",
      "  656 sequences unclassified (97.04%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4039.4 Kseq/m, 355.47 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2329.2 Kseq/m, 204.97 Mbp/m).\n",
      "  3 sequences classified (0.44%)\n",
      "  673 sequences unclassified (99.56%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2400.6 Kseq/m, 211.25 Mbp/m).\n",
      "  543 sequences classified (80.33%)\n",
      "  133 sequences unclassified (19.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3959.8 Kseq/m, 348.46 Mbp/m).\n",
      "  4 sequences classified (0.59%)\n",
      "  672 sequences unclassified (99.41%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.014s (2983.9 Kseq/m, 262.58 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4058.4 Kseq/m, 357.14 Mbp/m).\n",
      "  529 sequences classified (78.25%)\n",
      "  147 sequences unclassified (21.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2509.1 Kseq/m, 220.80 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2532.5 Kseq/m, 222.86 Mbp/m).\n",
      "  22 sequences classified (3.25%)\n",
      "  654 sequences unclassified (96.75%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2258.7 Kseq/m, 198.77 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3536.8 Kseq/m, 311.24 Mbp/m).\n",
      "  155 sequences classified (22.93%)\n",
      "  521 sequences unclassified (77.07%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.013s (3205.8 Kseq/m, 282.11 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2307.7 Kseq/m, 203.08 Mbp/m).\n",
      "  194 sequences classified (28.70%)\n",
      "  482 sequences unclassified (71.30%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.019s (2186.6 Kseq/m, 192.42 Mbp/m).\n",
      "  1 sequences classified (0.15%)\n",
      "  675 sequences unclassified (99.85%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2276.9 Kseq/m, 200.36 Mbp/m).\n",
      "  15 sequences classified (2.22%)\n",
      "  661 sequences unclassified (97.78%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2302.7 Kseq/m, 202.64 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2332.1 Kseq/m, 205.23 Mbp/m).\n",
      "  45 sequences classified (6.66%)\n",
      "  631 sequences unclassified (93.34%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.026s (1576.3 Kseq/m, 138.72 Mbp/m).\n",
      "  35 sequences classified (5.18%)\n",
      "  641 sequences unclassified (94.82%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2203.6 Kseq/m, 193.92 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2667.9 Kseq/m, 234.77 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3379.2 Kseq/m, 297.37 Mbp/m).\n",
      "  47 sequences classified (6.95%)\n",
      "  629 sequences unclassified (93.05%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2474.8 Kseq/m, 217.79 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3726.9 Kseq/m, 327.97 Mbp/m).\n",
      "  615 sequences classified (90.98%)\n",
      "  61 sequences unclassified (9.02%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2295.3 Kseq/m, 201.99 Mbp/m).\n",
      "  5 sequences classified (0.74%)\n",
      "  671 sequences unclassified (99.26%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2488.5 Kseq/m, 218.99 Mbp/m).\n",
      "  464 sequences classified (68.64%)\n",
      "  212 sequences unclassified (31.36%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2200.9 Kseq/m, 193.68 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2287.8 Kseq/m, 201.32 Mbp/m).\n",
      "  602 sequences classified (89.05%)\n",
      "  74 sequences unclassified (10.95%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.018s (2198.4 Kseq/m, 193.46 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.015s (2619.1 Kseq/m, 230.48 Mbp/m).\n",
      "  0 sequences classified (0.00%)\n",
      "  676 sequences unclassified (100.00%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.011s (3529.1 Kseq/m, 310.56 Mbp/m).\n",
      "  2 sequences classified (0.30%)\n",
      "  674 sequences unclassified (99.70%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3284.2 Kseq/m, 289.01 Mbp/m).\n",
      "  647 sequences classified (95.71%)\n",
      "  29 sequences unclassified (4.29%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.035s (1164.7 Kseq/m, 102.49 Mbp/m).\n",
      "  612 sequences classified (90.53%)\n",
      "  64 sequences unclassified (9.47%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.012s (3383.7 Kseq/m, 297.76 Mbp/m).\n",
      "  9 sequences classified (1.33%)\n",
      "  667 sequences unclassified (98.67%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (4028.2 Kseq/m, 354.48 Mbp/m).\n",
      "  110 sequences classified (16.27%)\n",
      "  566 sequences unclassified (83.73%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.010s (3962.5 Kseq/m, 348.70 Mbp/m).\n",
      "  615 sequences classified (90.98%)\n",
      "  61 sequences unclassified (9.02%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.017s (2363.6 Kseq/m, 208.00 Mbp/m).\n",
      "  603 sequences classified (89.20%)\n",
      "  73 sequences unclassified (10.80%)\n",
      "Loading database information... done.\n",
      "676 sequences (0.06 Mbp) processed in 0.016s (2485.9 Kseq/m, 218.76 Mbp/m).\n",
      "  203 sequences classified (30.03%)\n",
      "  473 sequences unclassified (69.97%)\n"
     ]
    }
   ],
   "source": [
    "for mut_file in glob.glob(f\"{out}/*_ms.fa\"):\n",
    "    out_file1 = mut_file.split(\"/\")[-1].split(\".fa\")[0] + \".k2report\"\n",
    "    out_file2 = mut_file.split(\"/\")[-1].split(\".fa\")[0] + \".kraken2\"\n",
    "    \n",
    "    !~/bin/kraken2 \\\n",
    "            --db $krakendb \\\n",
    "            --threads 32 \\\n",
    "            --report-minimizer-data \\\n",
    "            --report $kraken_out/$out_file1 \\\n",
    "            $mut_file > $kraken_out/$out_file2"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f27390b-57cc-4304-b8c7-ff5e67fc8dcd",
   "metadata": {},
   "source": [
    "### Align using kallisto bustools (translated search):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "d83127b9-3656-4b82-b178-0302a8576413",
   "metadata": {},
   "outputs": [],
   "source": [
    "kallisto_out = f\"{out}/../kallisto_mutations\"\n",
    "virus_t2g = \"/home/laura/projects/virus-watch-data/virus_ref/nodup_clu_t2g.txt\"\n",
    "virus_index = \"/home/laura/projects/virus-watch-data/virus_ref/kallisto_aa/noduplicates_rhesus.idx\"\n",
    "\n",
    "kallisto = \"/home/laura/projects/kallisto/build/src/kallisto\"\n",
    "bustools = \"/home/laura/projects/bustools/build/src/bustools\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "8b55300e-7524-494b-bc96-0f273362009d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_2_extracted_u10.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 676 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 676 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n"
     ]
    }
   ],
   "source": [
    "mut_file = \"/home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_2_extracted_u10.fa\"\n",
    "mut_name = \"0_ebov_snp_0_ms\"\n",
    "    \n",
    "!mkdir -p $kallisto_out/$mut_name\n",
    "\n",
    "!$kallisto bus \\\n",
    "    -i $virus_index \\\n",
    "    -o $kallisto_out/$mut_name \\\n",
    "    --aa \\\n",
    "    -x bulk \\\n",
    "    -t 32 \\\n",
    "    $mut_file\n",
    "\n",
    "!$bustools sort -o $kallisto_out/$mut_name/output_sorted.bus $kallisto_out/$mut_name/output.bus\n",
    "\n",
    "!$bustools count \\\n",
    "    --genecounts \\\n",
    "    --cm \\\n",
    "    -o $kallisto_out/$mut_name/bustools_count/ \\\n",
    "    -g $virus_t2g \\\n",
    "    -e $kallisto_out/$mut_name/matrix.ec \\\n",
    "    -t $kallisto_out/$mut_name/transcripts.txt \\\n",
    "    $kallisto_out/$mut_name/output_sorted.bus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "f6fd0948-88eb-48fd-8f18-2cbff8478b9b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 53 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 53 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 335 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 335 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 125 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 125 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 72 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 72 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 19 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 19 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 44 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 44 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 486 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 486 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 156 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 156 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 271 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 271 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 620 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 620 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 84 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 84 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 43 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 43 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 19 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 19 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 185 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 185 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 227 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 227 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 79 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 79 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 30 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 30 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 18 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 18 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 11 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 11 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 449 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 449 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 62 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 62 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 32 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 32 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 528 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 528 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 611 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 611 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 483 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 483 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 324 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 324 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 49 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 49 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 14 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 14 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 72 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 72 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 25 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 25 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 48 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 48 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 306 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 306 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 67 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 67 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 32 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 32 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 35 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 35 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 29 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 29 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 123 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 123 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 15 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 15 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 116 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 116 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 177 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 177 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 36 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 36 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 192 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 192 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 127 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 127 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 4 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 4 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 75 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 75 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 319 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 319 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 41 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 41 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 13 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 13 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 534 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 534 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 142 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 142 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 11 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 11 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 12 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 12 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 4 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 4 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 15 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 15 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 447 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 447 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 613 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 613 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 271 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 271 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 621 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 621 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 87 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 87 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 412 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 412 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 5 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 5 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 10 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 10 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 195 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 195 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 338 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 338 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 30 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 30 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 466 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 466 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 17 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 17 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 258 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 258 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 45 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 45 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 84 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 84 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 569 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 569 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 113 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 113 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 475 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 475 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 254 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 254 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 69 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 69 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 66 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 66 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 207 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 207 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 101 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 101 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 78 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 78 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 18 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 18 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 126 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 126 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 16 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 16 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 543 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 543 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 52 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 52 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 353 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 353 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 78 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 78 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 341 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 341 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 115 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 115 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 19 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 19 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 140 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 140 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 23 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 23 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 570 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 570 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 179 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 179 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 212 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 212 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 584 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 584 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 217 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 217 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 26 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 26 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 344 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 344 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 18 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 18 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 21 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 21 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 117 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 117 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 117 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 117 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 622 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 622 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 5 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 5 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 100 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 100 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 26 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 26 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 330 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 330 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 432 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 432 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 45 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 45 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 317 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 317 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 11 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 11 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 619 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 619 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 362 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 362 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 91 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 91 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 294 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 294 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 348 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 348 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 23 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 23 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 422 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 422 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 150 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 150 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 10 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 10 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 608 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 608 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 154 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 154 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 11 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 11 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 12 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 12 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 24 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 24 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 335 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 335 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 10 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 10 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 22 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 22 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 54 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 54 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 13 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 13 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 362 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 362 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 529 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 529 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 14 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 14 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 58 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 58 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 208 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 208 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 12 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 12 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 13 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 13 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 308 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 308 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 110 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 110 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 38 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 38 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 5 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 5 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 189 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 189 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 568 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 568 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0.01s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 73 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 73 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 498 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 498 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 541 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 541 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 195 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 195 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 427 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 427 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 64 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 64 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 4 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 4 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 385 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 385 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 12 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 12 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 32 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 32 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 442 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 442 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 75 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 75 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 490 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 490 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 342 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 342 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 239 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 239 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 306 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 306 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 186 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 186 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 13 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 13 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 71 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 71 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 370 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 370 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 81 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 81 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 265 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 265 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 96 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 96 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 262 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 262 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 21 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 21 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 11 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 11 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 204 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 204 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 316 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 316 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 283 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 283 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 161 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 161 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 238 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 238 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 20 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 20 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 77 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 77 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 442 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 442 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 80 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 80 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 150 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 150 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 310 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 310 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 621 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 621 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 480 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 480 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 10 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 10 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 110 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 110 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 464 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 464 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 84 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 84 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 408 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 408 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 82 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 82 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 37 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 37 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 12 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 12 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 3 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 3 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 76 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 76 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 85 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 85 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 17 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 17 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 61 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 61 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 41 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 41 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 11 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 11 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 273 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 273 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 351 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 351 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 123 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 123 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 233 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 233 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 535 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 535 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 23 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 23 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 3 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 3 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 177 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 177 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 3 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 3 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 255 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 255 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 33 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 33 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 357 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 357 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 531 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 531 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 471 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 471 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 318 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 318 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 414 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 414 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 29 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 29 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 73 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 73 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 26 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 26 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 146 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 146 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 43 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 43 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 522 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 522 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 235 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 235 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 570 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 570 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 28 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 28 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 79 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 79 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 15 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 15 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 23 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 23 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 318 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 318 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 26 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 26 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 315 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 315 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 104 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 104 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 373 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 373 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 597 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 597 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 96 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 96 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 14 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 14 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 24 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 24 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 520 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 520 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 42 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 42 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 12 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 12 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 531 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 531 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 87 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 87 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 263 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 263 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 317 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 317 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 13 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 13 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 68 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 68 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 162 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 162 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 101 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 101 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 18 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 18 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 174 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 174 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 564 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 564 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 32 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 32 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 501 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 501 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 548 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 548 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 10 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 10 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 20 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 20 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 38 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 38 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 623 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 623 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 571 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 571 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 54 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 54 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 217 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 217 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 568 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 568 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 564 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 564 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 296,561\n",
      "[index] number of k-mers: 37,541,842\n",
      "[index] number of distinguishing flanking k-mers: 138\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 336 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 336 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n"
     ]
    }
   ],
   "source": [
    "for mut_file in glob.glob(f\"{out}/*_ms.fa\"):\n",
    "    mut_name = mut_file.split(\"/\")[-1].split(\".fa\")[0]\n",
    "    \n",
    "    !mkdir -p $kallisto_out/$mut_name\n",
    "    \n",
    "    !$kallisto bus \\\n",
    "        -i $virus_index \\\n",
    "        -o $kallisto_out/$mut_name \\\n",
    "        --aa \\\n",
    "        -x bulk \\\n",
    "        -t 32 \\\n",
    "        $mut_file\n",
    "\n",
    "    !$bustools sort -o $kallisto_out/$mut_name/output_sorted.bus $kallisto_out/$mut_name/output.bus\n",
    "\n",
    "    !$bustools count \\\n",
    "        --genecounts \\\n",
    "        --cm \\\n",
    "        -o $kallisto_out/$mut_name/bustools_count/ \\\n",
    "        -g $virus_t2g \\\n",
    "        -e $kallisto_out/$mut_name/matrix.ec \\\n",
    "        -t $kallisto_out/$mut_name/transcripts.txt \\\n",
    "        $kallisto_out/$mut_name/output_sorted.bus"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a9efe25-a9a8-402b-872c-3fb896fa4527",
   "metadata": {},
   "source": [
    "### Align using kallisto bustools (standard workflow):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "24a7b97e-1772-4b77-b9db-9690f8eefc1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "kallisto_out = f\"{out}/../kallisto_mutations/nn\"\n",
    "ebola_t2g = \"/home/laura/projects/virus-watch-data/ebola_ref/ebov_GCA_000848505_t2g.txt\"\n",
    "ebola_index = \"/home/laura/projects/virus-watch-data/ebola_ref/ebov_GCA_000848505.idx\"\n",
    "\n",
    "kallisto = \"/home/laura/projects/kallisto/build/src/kallisto\"\n",
    "bustools = \"/home/laura/projects/bustools/build/src/bustools\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "74836649-46ca-49d9-b17e-a20e3ffcd8d4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_2_extracted_u10.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 662 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 662 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n"
     ]
    }
   ],
   "source": [
    "mut_file = \"/home/laura/projects/virus-watch-data/PRJNA665227/raw/SRR12698539_2_extracted_u10.fa\"\n",
    "mut_name = \"0_ebov_snp_0_ms\"\n",
    "    \n",
    "!mkdir -p $kallisto_out/$mut_name\n",
    "\n",
    "!$kallisto bus \\\n",
    "    -i $ebola_index \\\n",
    "    -o $kallisto_out/$mut_name \\\n",
    "    -x bulk \\\n",
    "    -t 32 \\\n",
    "    $mut_file\n",
    "\n",
    "!$bustools sort -o $kallisto_out/$mut_name/output_sorted.bus $kallisto_out/$mut_name/output.bus\n",
    "\n",
    "!$bustools count \\\n",
    "    --genecounts \\\n",
    "    --cm \\\n",
    "    -o $kallisto_out/$mut_name/bustools_count/ \\\n",
    "    -g $ebola_t2g \\\n",
    "    -e $kallisto_out/$mut_name/matrix.ec \\\n",
    "    -t $kallisto_out/$mut_name/transcripts.txt \\\n",
    "    $kallisto_out/$mut_name/output_sorted.bus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "5e71bde7-2519-43bf-9477-e8868af0dddb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 174 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 174 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 24 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 24 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 5 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 5 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 399 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 399 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 30 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 30 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 106 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 106 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 638 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 638 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 40 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 40 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 78 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 78 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 305 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 305 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 3 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 3 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 480 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 480 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 635 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 635 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 401 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 401 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 140 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 140 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 4 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 4 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 3 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 3 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 139 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 139 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 20 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 20 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 12 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 12 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 48 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 48 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 45 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 45 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 22 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 22 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 3 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 3 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 174 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 174 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0.02s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 475 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 475 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 31 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 31 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 295 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 295 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 637 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 637 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 96 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 96 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 638 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 638 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 280 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 280 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 50 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 50 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 148 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 148 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 383 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 383 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 100 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 100 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 5 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 5 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 13 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 13 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 578 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 578 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 14 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 14 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 379 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 379 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 103 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 103 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 11 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 11 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 65 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 65 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 18 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 18 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 30 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 30 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 500 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 500 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 197 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 197 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 142 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 142 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 17 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 17 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 23 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 23 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 588 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 588 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 54 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 54 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 78 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 78 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 583 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 583 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 74 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 74 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 139 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 139 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 17 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 17 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 15 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 15 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 639 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 639 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 12 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 12 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 159 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 159 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 288 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 288 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 157 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 157 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 631 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 631 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 217 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 217 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 134 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 134 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 207 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 207 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 303 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 303 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 34 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 34 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 637 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 637 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 36 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 36 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 3 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 3 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 144 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 144 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 207 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 207 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 498 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 498 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 5 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 5 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 64 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 64 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 144 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 144 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 15 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 15 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 46 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 46 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 581 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 581 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 366 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 366 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 477 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 477 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 63 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 63 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 305 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 305 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 210 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 210 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 325 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 325 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 7 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 7 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 410 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 410 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 214 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 214 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 76 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 76 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 141 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 141 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 52 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 52 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 217 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 217 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 14 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 14 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 114 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 114 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 15 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 15 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 93 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 93 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 48 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 48 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 137 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 137 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 117 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 117 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 29 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 29 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 62 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 62 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 275 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 275 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 12 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 12 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 35 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 35 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 149 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 149 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 636 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 636 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 395 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 395 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 19 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 19 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 361 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 361 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 274 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 274 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 6 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 6 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 8 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 8 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_18_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 4 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 4 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_29_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 94 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 94 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 203 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 203 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_14_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 26 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 26 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 65 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 65 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 485 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 485 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_12_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 52 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 52 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 101 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 101 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 3 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 3 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 221 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 221 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 467 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 467 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 374 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 374 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 131 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 131 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_06_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 270 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 270 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 5 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 5 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 38 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 38 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 2 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 2 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 479 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 479 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 62 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 62 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 577 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 577 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_17_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 11 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 11 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_23_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 140 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 140 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_21_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 124 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 124 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 16 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 16 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_07_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 207 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 207 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 636 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 636 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 13 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 13 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_24_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 482 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 482 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_04_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 476 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 476 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 10 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 10 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_1_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 122 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 122 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_09_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 146 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 146 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_26_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_16_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 5 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 5 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 31 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 31 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_15_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 16 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 16 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_25_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_13_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 43 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 43 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_3_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/5_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 586 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 586 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/6_ebov_snp_05_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 398 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 398 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_28_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/2_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 573 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 573 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_27_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_22_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
      "\n",
      "\n",
      "Read in 0 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/8_ebov_snp_2_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 1 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 1 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_02_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 632 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 632 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/3_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 577 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 577 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/0_ebov_snp_19_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 9 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 9 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/4_ebov_snp_11_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 68 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 68 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/1_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 571 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 571 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/9_ebov_snp_03_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 574 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 574 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n",
      "\n",
      "[index] k-mer length: 31\n",
      "[index] number of targets: 38\n",
      "[index] number of k-mers: 17,819\n",
      "[quant] running in single-end mode\n",
      "[quant] will process file 1: /home/laura/projects/virus-watch-data/benchmarking/ebola_benchmark/mutated_files/7_ebov_snp_08_ms.fa\n",
      "[quant] finding pseudoalignments for all files ... done\n",
      "[quant] processed 676 reads, 139 reads pseudoaligned\n",
      "\n",
      " all fits in buffer\n",
      "Read in 139 BUS records\n",
      "reading time 0s\n",
      "sorting time 0s\n",
      "writing time 0s\n"
     ]
    }
   ],
   "source": [
    "for mut_file in glob.glob(f\"{out}/*_ms.fa\"):\n",
    "    mut_name = mut_file.split(\"/\")[-1].split(\".fa\")[0]\n",
    "    \n",
    "    !mkdir -p $kallisto_out/$mut_name\n",
    "    \n",
    "    !$kallisto bus \\\n",
    "        -i $ebola_index \\\n",
    "        -o $kallisto_out/$mut_name \\\n",
    "        -x bulk \\\n",
    "        -t 32 \\\n",
    "        $mut_file\n",
    "\n",
    "    !$bustools sort -o $kallisto_out/$mut_name/output_sorted.bus $kallisto_out/$mut_name/output.bus\n",
    "\n",
    "    !$bustools count \\\n",
    "        --genecounts \\\n",
    "        --cm \\\n",
    "        -o $kallisto_out/$mut_name/bustools_count/ \\\n",
    "        -g $ebola_t2g \\\n",
    "        -e $kallisto_out/$mut_name/matrix.ec \\\n",
    "        -t $kallisto_out/$mut_name/transcripts.txt \\\n",
    "        $kallisto_out/$mut_name/output_sorted.bus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f497de2-2cf5-4413-8842-3a6933c18632",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
